<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="urwahnawaz" />

<meta name="date" content="2025-02-27" />

<title>Calculation of mRNA-protein correlations across pan-cancer datasets</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">OnCorr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="calculate_correlations.html">Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="publication-figures.html">Main publication</a>
    </li>
    <li>
      <a href="supplementary-publication-figures.html">Supplementary</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/urwahnawaz/OnCorr">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Calculation of mRNA-protein correlations
across pan-cancer datasets</h1>
<h4 class="author">urwahnawaz</h4>
<h4 class="date">2025-02-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>OnCorr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250619code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250619)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250619code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250619)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-13
</li>
<li>
unnamed-chunk-14
</li>
<li>
unnamed-chunk-15
</li>
<li>
unnamed-chunk-16
</li>
<li>
unnamed-chunk-17
</li>
<li>
unnamed-chunk-18
</li>
<li>
unnamed-chunk-19
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-20
</li>
<li>
unnamed-chunk-21
</li>
<li>
unnamed-chunk-22
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>calculate_correlations_cache</code> and re-run the analysis. To
have workflowr automatically delete the cache directory prior to
building the file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazOnCorrtreec1db52b14778f2a74d442005c3a25d98ee344e12targetblankc1db52ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/urwahnawaz/OnCorr/tree/c1db52b14778f2a74d442005c3a25d98ee344e12" target="_blank">c1db52b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazOnCorrtreec1db52b14778f2a74d442005c3a25d98ee344e12targetblankc1db52ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/urwahnawaz/OnCorr/tree/c1db52b14778f2a74d442005c3a25d98ee344e12" target="_blank">c1db52b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/calculate_correlations_cache/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/CPTAC-data-correlations.Rmd
    Untracked:  data/Datasets/
    Untracked:  data/OncoKB/
    Untracked:  data/Pathways/
    Untracked:  data/Ranks/
    Untracked:  output/CPTAC-RNAseq-merged.csv
    Untracked:  output/CPTAC-proteomics-merged.csv
    Untracked:  output/CPTAC-sample-info-merged.csv
    Untracked:  output/Other/
    Untracked:  output/Publication/
    Untracked:  output/correlations-datasets/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/calculate_correlations.Rmd</code>) and HTML
(<code>docs/calculate_correlations.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/c1db52b14778f2a74d442005c3a25d98ee344e12/analysis/calculate_correlations.Rmd" target="_blank">c1db52b</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-30
</td>
<td>
wflow_publish(c("analysis/calculate_correlations.Rmd",
"analysis/index.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/OnCorr/ddc48e4cf88b0af734aeff7a869bc2db32f054dd/docs/calculate_correlations.html" target="_blank">ddc48e4</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/6b1840fd5854a5928c250096c415b1358688f9db/analysis/calculate_correlations.Rmd" target="_blank">6b1840f</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-24
</td>
<td>
wflow_publish(c("analysis/index.Rmd",
"analysis/calculate_correlations.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This notebook contains the code that calculates mRNA-protein
correlations for the 949 ProCan-DepMapSanger cancer cell line data. In
addition, it also contains code which calculates correlations for the
CPTAC data retrieved from LinkedOmics KB, and CCLE datasets.</p>
<p>We have calculated both correlations of the entire dataset, and
tissue specific correlations.</p>
<div id="libraries-used" class="section level3">
<h3>Libraries used</h3>
<pre class="r"><code>library(ggplot2)
library(magrittr)
library(tidyverse)
library(readxl)
library(dplyr)
library(RWDataPlyr)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(EnsDb.Hsapiens.v86)
library(svglite)
library(ggiraph)
library(reshape2)
library(ggvenn)
library(VennDiagram)
library(ggtext)
library(colorspace)
library(ragg)
library(msigdbr)
library(matrixStats)
library(ggiraph)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
</div>
<div id="procan-depmapsanger-dataset" class="section level1">
<h1>ProCan-DepMapSanger Dataset</h1>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r"><code>file_dir = file.path(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/&quot;)
data = list()
data$sample_md = read.delim(&quot;~/Documents/2023/RNA-protein-correlationx/Data/e0022_diann_051021_sample_mapping_averaged.txt&quot;)
data$sample_md_rep = read.table(file.path(file_dir, &quot;e0022_diann_051021_sample_mapping_replicates.txt&quot;), header=TRUE, sep = &quot;\t&quot;)

data$prt_mtx =  read.delim(file.path(file_dir, &quot;e0022_diann_051021_working_matrix_averaged.txt&quot;), 
                           header = TRUE, fill = TRUE, check.names = FALSE) 
data$rna_voom = read.csv(gzfile(file.path(file_dir, &quot;rnaseq_voom.csv.gz&quot;))) %&gt;%
  dplyr::rename(&quot;Gene&quot; = &quot;X&quot;)
uniprot_annot = read.table(file.path(file_dir, &quot;idmapping_2024_02_13.tsv&quot;), 
                           sep = &quot;\t&quot;, fill = TRUE, header=TRUE, quote = &quot;&quot;) %&gt;% 
  mutate(&quot;Protein_name&quot; = paste0(Entry, &quot;;&quot;, `Entry.Name`)) %&gt;% 
  distinct(Protein_name, .keep_all = TRUE) %&gt;% 
  dplyr::select(Protein_name, Gene = `Gene.Names`) %&gt;% 
   mutate(Gene = str_split(Gene, &quot; &quot;)) %&gt;%
  unnest(Gene)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>data$rna_voom %&lt;&gt;% 
  left_join(uniprot_annot , by = &quot;Gene&quot;) </code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="pan-cancer-correlations-in-the-procan-depmapsanger-dataset"
class="section level2">
<h2>Pan-cancer correlations in the ProCan-DepMapSanger dataset</h2>
<pre class="r"><code>data_full = data$prt_mtx %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;Project_Identifier&quot;) %&gt;%
  t() %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Protein&quot;) %&gt;%
 mutate(&quot;Protein_name&quot; = gsub(&quot;;.*&quot;, &quot;&quot;, Protein)) %&gt;%
  mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = &quot;UNIPROT&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;))</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>cor_all= list()
cell_lines = data$sample_md %&gt;% 
    dplyr::select(Project_Identifier, SIDM)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Proteins were considered if they had &gt; 10 observations for them in
each RNA and proteomic datasets.</p>
<pre class="r"><code>for (prt in data_full$Gene) {

 
    cor_df = data$rna_voom %&gt;% 
      dplyr::filter(Gene %in% prt) %&gt;% 
      melt() %&gt;% dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;RNA&quot; = &quot;value&quot;) %&gt;% 
      left_join(cell_lines, by =&quot;SIDM&quot;) %&gt;% 
      dplyr::select(&quot;Cell-lines&quot; = &quot;Project_Identifier&quot;, 
                    RNA) %&gt;%
      full_join(data_full %&gt;% 
        dplyr::filter(Gene %in% prt) %&gt;% 
          
      dplyr::select(contains(cell_lines$Project_Identifier)) %&gt;%
        melt() %&gt;% dplyr::rename(&quot;Cell-lines&quot; = &quot;variable&quot;, &quot;Protein&quot; = &quot;value&quot;), 
      by = &quot;Cell-lines&quot;) %&gt;% drop_na()
    
      if(nrow(cor_df) &gt; 10) { 

      cor_all[[paste0(prt)]]  = cor_df %&gt;% 
       with(cor(.$RNA,.$Protein, method = &quot;spearman&quot;)) %&gt;% 
        as.data.frame() %&gt;% 
        set_colnames(&quot;Correlation&quot;) %&gt;% 
        mutate(Protein = prt)

    }
    
}

corAll = cor_all %&gt;%
  do.call(rbind,.) %&gt;% 
  drop_na()



#write.csv(corAll, &quot;output/correlations-datasets/correlation-all-nontransformed.csv&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>corAll  %&gt;%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.1, y = 600,
  label = &quot;paste(italic(median), \&quot; = 0.42\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept = 0.42, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;ProCan-DepMapSanger&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/calculate_correlations.Rmd/unnamed-chunk-7-1.png" alt="Pan-cancer mRNA-protein correlations in the ProCan-DepMapSanger dataset." width="480" />
<p class="caption">
Pan-cancer mRNA-protein correlations in the ProCan-DepMapSanger dataset.
</p>
</div>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div
id="cancer-specific-correlations-in-the-procan-depmapsanger-dataset"
class="section level2">
<h2>Cancer-specific correlations in the ProCan-DepMapSanger dataset</h2>
<p>To increase statstical power and for consistency, we grouped cancers
based on their tissue of origin, We selected tissues which contain more
than 10 cell lines. Subsequently, only calculating correlations for
proteins which contain more than 10 observations for both RNA and
protein abundances.</p>
<pre class="r"><code>tissue_counts &lt;-table(data$sample_md$Tissue_type)
tissues_incl &lt;- names(tissue_counts[tissue_counts &gt; 10])

tissue_counts %&gt;% 
  as.data.frame() %&gt;%
  mutate(col = ifelse(Freq &gt;10, &quot;Keep&quot;, &quot;Not Keep&quot;)) %&gt;%
  ggplot(aes(x=reorder(Var1, Freq), y=Freq, fill = col)) + geom_bar(stat =&quot;identity&quot;)  + theme_bw() +
  coord_flip() + 
  labs(x=&quot;Tissues&quot;, y =&quot;Number of samples&quot;, 
       title = &quot;Tissues&quot;) +
   theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=10,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=12, family = &quot;serif&quot;, color = &quot;black&quot;)) + 
  scale_fill_manual(values = c(&quot;Keep&quot; = &quot;#8c2f39&quot;,
                               &quot;Not Keep &quot;))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/calculate_correlations.Rmd/unnamed-chunk-8-1.png" alt="Selecting tissues with &gt; 10 cell lines. Selected tissues are denoted in red, whereas those that do not meet the threshold are in grey." width="576" />
<p class="caption">
Selecting tissues with &gt; 10 cell lines. Selected tissues are denoted
in red, whereas those that do not meet the threshold are in grey.
</p>
</div>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>cor_per_tissue = list()


for (tis in tissues_incl){
  
  message(paste(&quot;Now performing correlation analysis for&quot;, tis))
  cell_lines = data$sample_md %&gt;% 
    dplyr::filter(Tissue_type ==  tis ) %&gt;% 
    dplyr::select(Project_Identifier, SIDM)
  
  cor_analysis = list()
  
  for (prt in data_full$Gene) {
 
    cor_df = data$rna_voom %&gt;% 
      dplyr::filter(Gene %in% prt) %&gt;% 
      dplyr::select(contains(cell_lines$SIDM)) %&gt;%
      melt() %&gt;% dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;RNA&quot; = &quot;value&quot;) %&gt;% 
      left_join(cell_lines, by =&quot;SIDM&quot;) %&gt;% 
      dplyr::select(&quot;Cell-lines&quot; = &quot;Project_Identifier&quot;, 
                    RNA) %&gt;%
      full_join(data_full%&gt;% 
                  dplyr::filter(Gene %in% prt)  %&gt;% 
                  dplyr::select(contains(cell_lines$Project_Identifier)) %&gt;%
                  melt() %&gt;% 
                  dplyr::rename(&quot;Cell-lines&quot; = &quot;variable&quot;, &quot;Protein&quot; = &quot;value&quot;),
      by = &quot;Cell-lines&quot;) %&gt;% drop_na()
    
      if(nrow(cor_df) &gt; 10) { 

      cor_per_tissue[[paste0(tis,&quot;_&quot;, prt)]]  = cor_df %&gt;% 
       with(cor(.$RNA,.$Protein, method = &quot;spearman&quot;)) %&gt;% 
        as.data.frame() %&gt;% 
        set_colnames(&quot;Correlation&quot;) %&gt;% 
        mutate(Tissue = tis, 
               Protein = prt)

    }
    
  }

}


corTissue = cor_per_tissue  %&gt;%
  do.call(rbind,.) %&gt;% 
  drop_na() %&gt;% 
  dplyr::rename(Gene = Protein) %&gt;% 
  left_join(data_full %&gt;% 
              dplyr::select(Gene, Protein), by = &quot;Gene&quot;)




#prev_tis_cor = read.csv(&quot;output/correlation-tissue-nontransformed.csv&quot;)

#prev_tis_cor %&gt;% DT::datatable()
#write.csv(corTissue, &quot;output/correlations-datasets/correlation-tissue-nontransformed.csv&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
</div>
<div id="cptac-dataset" class="section level1">
<h1>CPTAC dataset</h1>
<pre class="r"><code>subfolders = list.dirs(file.path(&quot;/Users/unawaz/Documents/2024/Correlations/CPTAC&quot;), full.names = TRUE)
rna_file = list.files(subfolders, pattern = &quot;RSEM&quot;, full.names = TRUE)
ptn_file = list.files(subfolders, pattern = &quot;normalized_Tumor.txt&quot;, full.names = TRUE)
metadata  =  list.files(subfolders, pattern = &quot;meta.txt&quot;, full.names = TRUE)
RNA_seq = list()

for (rna in rna_file){
  dataset_name = basename(dirname(rna))
  RNA_seq[[paste0(dataset_name)]] = read.table(rna, header=TRUE, check.names  = FALSE)

}


ptx = list()
for (ptn in ptn_file){
  dataset_name = basename(dirname(ptn))
  ptx[[paste0(dataset_name)]] = read.table(ptn, header=TRUE, check.names  = FALSE, row.names = 1)

}

md = list()

for (met in metadata){
  dataset_name = basename(dirname(met))
  md[[paste0(dataset_name)]] = read.table(met, check.names  = FALSE, fill = TRUE, header=TRUE, sep = &quot;\t&quot;)

}</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>We merged all the RNA and protein matrices into single data list for
each type of abundance across all cancers.</p>
<pre class="r"><code>cancers = names(ptx)

all_data = ptx$BRCA %&gt;%
  rownames_to_column(&quot;Gene&quot;)
for (cancer in cancers[-1]){

  all_data %&lt;&gt;%
    full_join(ptx[[cancer]] %&gt;%
                rownames_to_column(&quot;Gene&quot;), by = &quot;Gene&quot;)
}


#write.csv(all_data, &quot;output/CPTAC-proteomics-merged.csv&quot;)

all_data_rna = RNA_seq$BRCA

for (cancer in cancers[-1]){

  all_data_rna %&lt;&gt;%
    full_join(RNA_seq[[cancer]], by = &quot;idx&quot;)
}


#write.csv(all_data_rna, &quot;output/CPTAC-RNAseq-merged.csv&quot;)


library(data.table)

all_md = rbindlist(md, fill = TRUE) %&gt;% as.data.frame()
#write.csv(all_md, &quot;output/CPTAC-sample-info-merged.csv&quot;)



for (cancer in cancers){

  md[[cancer]] %&lt;&gt;%
    mutate(Cancer = cancer)
}</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div id="pan-cancer-correlations-in-the-cptac-dataset"
class="section level2">
<h2>Pan-cancer correlations in the CPTAC dataset</h2>
<p>Similar to the ProCan-DepMapSanger dataset, we performed correlations
across the entire dataset, where only proteins with &gt; 10 observations
in both the protein matrix and the RNA matrix were considered for
analysis.</p>
<pre class="r"><code>cor_all = list()

for (protein in all_data$Gene) {


cor_df =  all_data_rna %&gt;%
      dplyr::filter(idx %in% protein) %&gt;%
      melt() %&gt;% dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;RNA&quot; = &quot;value&quot;) %&gt;%
      full_join(all_data %&gt;%
                   dplyr::filter(Gene %in% protein) %&gt;%
                  melt() %&gt;%
      dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;Protein&quot; = &quot;value&quot;) %&gt;%
        dplyr::select(SIDM, Protein),
      by = &quot;SIDM&quot;) %&gt;% drop_na()

      if(nrow(cor_df) &gt; 10) {

      cor_all[[paste0(protein)]]  = cor_df %&gt;%
       with(cor(.$RNA,.$Protein, method = &quot;spearman&quot;)) %&gt;%
        as.data.frame() %&gt;%
        set_colnames(&quot;Correlation&quot;) %&gt;%
        mutate(
               Protein = protein)

    }

}


cptac=  cor_all %&gt;%
  do.call(rbind,.)


cptac  %&gt;%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.1, y = 600,
  label = &quot;paste(italic(median), \&quot; = 0.39\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept = 0.39, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;CPTAC&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) </code></pre>
<div class="figure" style="text-align: center">
<img src="figure/calculate_correlations.Rmd/unnamed-chunk-12-1.png" alt="Pan-cancer mRNA-protein correlations in the CPTAC dataset." width="480" />
<p class="caption">
Pan-cancer mRNA-protein correlations in the CPTAC dataset.
</p>
</div>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="cancer-specific-correlations-in-the-cptac-dataset"
class="section level2">
<h2>Cancer-specific correlations in the CPTAC dataset</h2>
<p>Similar to the ProCan-DepMap sanger dataset, we grouped cancers based
on their tissue of origin, and selected tissues which contain more than
10 cell lines.</p>
<pre class="r"><code>cor_per_cancer = list()
cancers = names(ptx)

for (cancer in cancers){
  ptx_matrix = ptx[[cancer]]
  rna_matrix = RNA_seq[[cancer]]


   for (protein in rownames(ptx_matrix)) {

    cor_df = rna_matrix %&gt;%
      dplyr::filter(idx %in% protein) %&gt;%
      melt() %&gt;% dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;RNA&quot; = &quot;value&quot;) %&gt;%
      full_join(ptx_matrix %&gt;%
                  rownames_to_column(&quot;Genes&quot;) %&gt;%
                   dplyr::filter(Genes %in% protein) %&gt;%
                  melt() %&gt;%
      dplyr::rename(&quot;SIDM&quot; = &quot;variable&quot;, &quot;Protein&quot; = &quot;value&quot;) %&gt;%
        dplyr::select(SIDM, Protein),
      by = &quot;SIDM&quot;) %&gt;% drop_na()

      if(nrow(cor_df) &gt; 10) {

      cor_per_cancer[[paste0(cancer,&quot;_&quot;, protein)]]  = cor_df %&gt;%
       with(cor(.$RNA,.$Protein, method = &quot;spearman&quot;)) %&gt;%
        as.data.frame() %&gt;%
        set_colnames(&quot;Correlation&quot;) %&gt;%
        mutate(Cancer = cancer,
               Protein = protein)

    }

  }
}

cptac= cor_per_cancer %&gt;%
  do.call(rbind,.)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
</div>
<div id="ccle-dataset" class="section level1">
<h1>CCLE dataset</h1>
<pre class="r"><code>CCLE_data = read_excel(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/mmc4.xlsx&quot;,
                       sheet =2)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>data_ccle = list()

data_ccle$prt = read.csv(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/protein_quant_current_normalized.csv&quot;, header=TRUE)
data_ccle$rna = read.csv(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/CCLE_expression.csv&quot;,
                         header=TRUE, check.names = FALSE)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>data_ccle$sample_md = read.csv(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/sample_info.csv&quot;, header=TRUE,
                               check.names = FALSE)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div id="fix-metadata-for-tissues" class="section level2">
<h2>Fix metadata for tissues</h2>
<pre class="r"><code>ccle_tissue_annot = read_excel(&quot;data/Datasets/CCLE/CCLE_tissue_names.xlsx&quot;) %&gt;%
  as.data.frame()</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>data_ccle$sample_md %&lt;&gt;%
  mutate(Lineage = gsub(&quot;_&quot;, &quot; &quot;, lineage)) %&gt;%
  mutate(Lineage = str_to_title(Lineage)) %&gt;%
  left_join(ccle_tissue_annot, by = &quot;Lineage&quot;)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>colnames(data_ccle$rna)[1] = c(&quot;DepMap_ID&quot;)

data_ccle$rna %&lt;&gt;%
  column_to_rownames(&quot;DepMap_ID&quot;) %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Gene&quot;) %&gt;%
  mutate(Gene = gsub(&quot; .*&quot;, &quot;&quot;, Gene))  %&gt;%
  mutate(&quot;Protein&quot; = mapIds(org.Hs.eg.db, keys = Gene,
                         column = &quot;UNIPROT&quot;, keytype= &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;%
  drop_na(Protein)


data_ccle$prt = data_ccle$prt[,-c(7:48)]


data_ccle$prt %&lt;&gt;%
  dplyr::select(-c(Protein_Id, Description,
                &quot;Group_ID&quot;,&quot;Uniprot&quot;)) %&gt;%
  column_to_rownames(&quot;Uniprot_Acc&quot;)

colnames(data_ccle$prt) = gsub(&quot;_Ten.*&quot;, &quot;&quot;,colnames(data_ccle$prt))</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<pre class="r"><code>library(ggvenn)
library(VennDiagram)

x = list(&quot;Protein&quot; = data_ccle$prt$Gene_Symbol,
         &quot;Gene&quot; = data_ccle$rna$Gene)

lists = calculate.overlap(x)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<div id="correlation-across-the-dataset" class="section level3">
<h3>Correlation across the dataset</h3>
<pre class="r"><code>ccle_cor_all  = list()



prt_data = data_ccle$prt %&gt;%
  melt() %&gt;%
  dplyr::rename(CCLE_Name = variable, Protein = value) %&gt;%
  inner_join(data_ccle$sample_md %&gt;%
               dplyr::select(CCLE_Name, DepMap_ID,  `Tissue_type`= `New Lineage`), by = &quot;CCLE_Name&quot;) %&gt;%
  dplyr::select(-c(CCLE_Name))

 rna_data = data_ccle$rna %&gt;%
   melt() %&gt;%
   dplyr::select(-Protein) %&gt;%
   dplyr::rename(DepMap_ID = variable, &quot;RNA&quot; = value)

 ccle_exp_data = rna_data %&gt;%
   mutate(identifier = paste0(Gene, &quot;_&quot;, DepMap_ID)) %&gt;%
   distinct(identifier, .keep_all = TRUE) %&gt;%
   dplyr::select(- DepMap_ID) %&gt;%
   left_join(prt_data %&gt;%
               mutate(identifier = paste0(Gene_Symbol, &quot;_&quot;, DepMap_ID)),  by = &quot;identifier&quot;) %&gt;%
   drop_na() %&gt;%
   dplyr::select(-Gene_Symbol) %&gt;%
   dplyr::select(Gene, SIDM= DepMap_ID, RNA, Protein, Tissue_type) %&gt;%
   mutate(Dataset = c(&quot;CCLE&quot;),
          Tissue_type = ifelse(Tissue_type == &quot;Lubg&quot;, &quot;Lung&quot;, Tissue_type))

  for (prt in lists$a3) {

    cor_data= rna_data %&gt;%
      dplyr::filter(Gene == prt) %&gt;%
      full_join(prt_data %&gt;%
                  dplyr::filter(Gene_Symbol == prt) %&gt;%
                  dplyr::select(-Gene_Symbol),
                by = &quot;DepMap_ID&quot;) %&gt;%
      drop_na()

      if(nrow(cor_data) &gt; 10) {

       ccle_cor_all[[paste(prt)]]  = cor_data %&gt;%
         with(cor(.$Protein, .$RNA, method = &quot;spearman&quot;)) %&gt;%
         as.data.frame() %&gt;%
        set_colnames(&quot;Correlation&quot;) %&gt;%
       mutate(Protein = prt)

    }

  }

 ccle_cor_all %&lt;&gt;% do.call(rbind,.) %&gt;%  drop_na()

 
ccle_cor_all %&gt;% 
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.1, y = 600,
  label = &quot;paste(italic(median), \&quot; = 0.45\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept = 0.45, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;CCLE&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) </code></pre>
<p><img src="figure/calculate_correlations.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code> median(ccle_cor_all$Correlation)</code></pre>
<pre><code>[1] 0.452425</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
<div id="tissue-type-specifc" class="section level3">
<h3>Tissue type specifc</h3>
<pre class="r"><code>cancers = table(data_ccle$sample_md$primary_disease) %&gt;%
   melt() %&gt;%
   dplyr::filter(value &gt;10) %&gt;%
   set_colnames(&quot;Cancer&quot;) %&gt;%
   as.data.frame()


tissues= colnames(data_ccle$prt) %&gt;%
  as.data.frame() %&gt;%
  set_names(c(&quot;CCLE_Name&quot;)) %&gt;%
  inner_join(data_ccle$sample_md, by = &quot;CCLE_Name&quot;) %&gt;%
  with(table(.$`New Lineage`)) %&gt;%
  melt() %&gt;%
  dplyr::filter(value &gt;10) %&gt;%
  set_colnames(c(&quot;Tissue&quot;, &quot;Number&quot;))

ccle_cor = list()
for (cancer in tissues$Tissue) {
  samples =  data_ccle$sample_md %&gt;%
  dplyr::filter(`New Lineage` %in% cancer)

  rna_data = data_ccle$rna %&gt;%
    dplyr::select(Gene, contains(samples$DepMap_ID))

  prt_data = data_ccle$prt %&gt;%
    dplyr::select(Gene_Symbol, any_of(samples$CCLE_Name)) %&gt;%
  melt() %&gt;%
  dplyr::rename(CCLE_Name = variable, Protein = value) %&gt;%
  inner_join(samples %&gt;%
               dplyr::select(CCLE_Name, DepMap_ID), by = &quot;CCLE_Name&quot;) %&gt;%
  dplyr::select(-c(CCLE_Name))

 rna_data = data_ccle$rna %&gt;%
    dplyr::select(Gene, any_of(samples$DepMap_ID)) %&gt;%
   melt() %&gt;%
   dplyr::rename(DepMap_ID = variable, &quot;RNA&quot; = value)

  for (prt in lists$a3) {

    cor_data= rna_data %&gt;%
      dplyr::filter(Gene == prt) %&gt;%
      full_join(prt_data %&gt;%
                  dplyr::filter(Gene_Symbol == prt) %&gt;%
                  dplyr::select(-Gene_Symbol),
                by = &quot;DepMap_ID&quot;) %&gt;%
      drop_na()

      if(nrow(cor_data) &gt; 10) {

       ccle_cor[[paste(prt, cancer)]]  = cor_data %&gt;%
         with(cor(.$Protein, .$RNA, method = &quot;spearman&quot;)) %&gt;%
         as.data.frame() %&gt;%
        set_colnames(&quot;Correlation&quot;) %&gt;%
       mutate(Tissue = cancer,
             Protein = prt)

    }

  }

}

ccle_cor %&lt;&gt;% do.call(rbind,.) %&gt;%  drop_na()</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Sydney
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] matrixStats_1.5.0         msigdbr_7.5.1            
 [3] ragg_1.3.3                colorspace_2.1-1         
 [5] ggtext_0.1.2              VennDiagram_1.7.3        
 [7] futile.logger_1.4.3       ggvenn_0.1.10            
 [9] reshape2_1.4.4            ggiraph_0.8.12           
[11] svglite_2.1.3             EnsDb.Hsapiens.v86_2.99.0
[13] ensembldb_2.26.0          AnnotationFilter_1.26.0  
[15] GenomicFeatures_1.54.4    GenomicRanges_1.54.1     
[17] GenomeInfoDb_1.38.8       org.Hs.eg.db_3.18.0      
[19] AnnotationDbi_1.64.1      IRanges_2.36.0           
[21] S4Vectors_0.40.2          Biobase_2.62.0           
[23] BiocGenerics_0.48.1       RWDataPlyr_0.6.4         
[25] readxl_1.4.3              lubridate_1.9.4          
[27] forcats_1.0.0             stringr_1.5.1            
[29] dplyr_1.1.4               purrr_1.0.2              
[31] readr_2.1.5               tidyr_1.3.1              
[33] tibble_3.2.1              tidyverse_2.0.0          
[35] magrittr_2.0.3            ggplot2_3.5.1            
[37] workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] rstudioapi_0.17.1           jsonlite_1.8.9             
  [3] farver_2.1.2                rmarkdown_2.29             
  [5] fs_1.6.5                    BiocIO_1.12.0              
  [7] zlibbioc_1.48.2             vctrs_0.6.5                
  [9] memoise_2.0.1               Rsamtools_2.18.0           
 [11] RCurl_1.98-1.16             htmltools_0.5.8.1          
 [13] S4Arrays_1.2.1              progress_1.2.3             
 [15] lambda.r_1.2.4              curl_6.1.0                 
 [17] cellranger_1.1.0            SparseArray_1.2.4          
 [19] sass_0.4.9                  bslib_0.8.0                
 [21] htmlwidgets_1.6.4           plyr_1.8.9                 
 [23] futile.options_1.0.1        cachem_1.1.0               
 [25] uuid_1.2-1                  GenomicAlignments_1.38.2   
 [27] whisker_0.4.1               lifecycle_1.0.4            
 [29] pkgconfig_2.0.3             Matrix_1.6-5               
 [31] R6_2.5.1                    fastmap_1.2.0              
 [33] GenomeInfoDbData_1.2.11     MatrixGenerics_1.14.0      
 [35] digest_0.6.37               ps_1.8.1                   
 [37] rprojroot_2.0.4             textshaping_0.4.1          
 [39] RSQLite_2.3.9               labeling_0.4.3             
 [41] filelock_1.0.3              timechange_0.3.0           
 [43] httr_1.4.7                  abind_1.4-8                
 [45] compiler_4.3.3              bit64_4.5.2                
 [47] withr_3.0.2                 BiocParallel_1.36.0        
 [49] DBI_1.2.3                   biomaRt_2.58.2             
 [51] rappdirs_0.3.3              DelayedArray_0.28.0        
 [53] rjson_0.2.23                tools_4.3.3                
 [55] httpuv_1.6.15               glue_1.8.0                 
 [57] restfulr_0.0.15             callr_3.7.6                
 [59] promises_1.3.2              gridtext_0.1.5             
 [61] getPass_0.2-4               generics_0.1.3             
 [63] gtable_0.3.6                tzdb_0.4.0                 
 [65] hms_1.1.3                   xml2_1.3.6                 
 [67] XVector_0.42.0              pillar_1.10.1              
 [69] babelgene_22.9              later_1.4.1                
 [71] BiocFileCache_2.10.2        lattice_0.22-5             
 [73] rtracklayer_1.62.0          bit_4.5.0.1                
 [75] tidyselect_1.2.1            Biostrings_2.70.3          
 [77] knitr_1.49                  git2r_0.35.0               
 [79] ProtGenerics_1.34.0         SummarizedExperiment_1.32.0
 [81] xfun_0.50                   stringi_1.8.4              
 [83] lazyeval_0.2.2              yaml_2.3.10                
 [85] evaluate_1.0.3              codetools_0.2-19           
 [87] cli_3.6.3                   systemfonts_1.1.0          
 [89] munsell_0.5.1               processx_3.8.5             
 [91] jquerylib_0.1.4             Rcpp_1.0.14                
 [93] dbplyr_2.5.0                png_0.1-8                  
 [95] XML_3.99-0.18               parallel_4.3.3             
 [97] blob_1.2.4                  prettyunits_1.2.0          
 [99] bitops_1.0-9                scales_1.3.0               
[101] crayon_1.5.3                rlang_1.1.4                
[103] formatR_1.14                KEGGREST_1.42.0            </code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
