<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="urwahnawaz" />

<meta name="date" content="2024-10-08" />

<title>Main publication figures for Nawaz et al</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">OnCorr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="calculate_correlations.html">Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="publication-figures.html">Main publication</a>
    </li>
    <li>
      <a href="supplementary-publication-figures.html">Supplementary</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/urwahnawaz/OnCorr">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Main publication figures for Nawaz et
al</h1>
<h4 class="author">urwahnawaz</h4>
<h4 class="date">2024-10-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-06-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OnCorr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250619code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250619)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250619code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250619)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazOnCorrtreead3a802a85ac1d60465e07afc324b2f423d1ddbbtargetblankad3a802a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/urwahnawaz/OnCorr/tree/ad3a802a85ac1d60465e07afc324b2f423d1ddbb" target="_blank">ad3a802</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazOnCorrtreead3a802a85ac1d60465e07afc324b2f423d1ddbbtargetblankad3a802a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/urwahnawaz/OnCorr/tree/ad3a802a85ac1d60465e07afc324b2f423d1ddbb" target="_blank">ad3a802</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/calculate_correlations_cache/
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/CPTAC-data-correlations.Rmd
    Untracked:  data/Datasets/
    Untracked:  data/OncoKB/
    Untracked:  data/Pathways/
    Untracked:  data/Ranks/
    Untracked:  output/CPTAC-RNAseq-merged.csv
    Untracked:  output/CPTAC-proteomics-merged.csv
    Untracked:  output/CPTAC-sample-info-merged.csv
    Untracked:  output/Other/
    Untracked:  output/Publication/
    Untracked:  output/correlations-datasets/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Modified:   analysis/calculate_correlations.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/publication-figures.Rmd</code>)
and HTML (<code>docs/publication-figures.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/ad3a802a85ac1d60465e07afc324b2f423d1ddbb/analysis/publication-figures.Rmd" target="_blank">ad3a802</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-25
</td>
<td>
wflow_publish(c("analysis/publication-figures.Rmd",
"analysis/index.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/OnCorr/0c31b7dfd8fe2fbcb3f9b55dab1234cb41578ebb/docs/publication-figures.html" target="_blank">0c31b7d</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/OnCorr/debfbfb5684aae59bf0ca220b356ac9b084d9d77/docs/publication-figures.html" target="_blank">debfbfb</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/OnCorr/339b94b804910fb54366c6dce4f1bfe0648655df/docs/publication-figures.html" target="_blank">339b94b</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/OnCorr/0462df41c2a09e5ded2a36493174bfe6c2de2586/docs/publication-figures.html" target="_blank">0462df4</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/9809e44e06972534d354fe8e26ce11a267bd3f22/analysis/publication-figures.Rmd" target="_blank">9809e44</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
<td>
wflow_publish("analysis/publication-figures.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This notebook contains code used to generate main figures in <a
href="https://www.biorxiv.org/content/10.1101/2025.06.12.659426v1">Nawaz
et al (2025)</a>. The code used to generate the mRNA-protein correlation
data used for these figures can be found in
<code>analysis/calculate_correlations.Rmd</code>, or in the Analysis
option on the option bar in this notebook. Figures which utilise the raw
protein or RNA expression matrices can be found in the relevant
publicaitons.</p>
<div id="import-libraries-and-set-up" class="section level2">
<h2>Import libraries and set-up</h2>
<pre class="r"><code>library(ggplot2)
library(magrittr)
library(tidyverse)
library(readxl)
library(dplyr)
library(RWDataPlyr)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(EnsDb.Hsapiens.v86)
library(svglite)
library(ggiraph)
library(reshape2)
library(ggvenn)
library(VennDiagram)
library(ggtext)
library(colorspace)
library(ragg)
library(msigdbr)
library(matrixStats)
library(ggpubr)</code></pre>
<pre class="r"><code>library(readxl)    
read_excel_allsheets &lt;- function(filename, tibble = FALSE) {
    # I prefer straight data.frames
    # but if you like tidyverse tibbles (the default with read_excel)
    # then just pass tibble = TRUE
    sheets &lt;- readxl::excel_sheets(filename)
    x &lt;- lapply(sheets, function(X) readxl::read_excel(filename, sheet = X))
    if(!tibble) x &lt;- lapply(x, as.data.frame)
    names(x) &lt;- sheets
    x
}</code></pre>
<pre class="r"><code>txdf = ensembldb::transcripts(EnsDb.Hsapiens.v86, return.type=&quot;DataFrame&quot;)
genedf = ensembldb::genes(EnsDb.Hsapiens.v86, return.type=&quot;DataFrame&quot;) %&gt;% 
  as.data.frame() %&gt;% 
  dplyr::select(gene_name, gene_id)
tx2gene = as.data.frame(txdf[,c(&quot;tx_id&quot;,&quot;gene_id&quot;, &quot;tx_biotype&quot;)])

 

tx2gene %&lt;&gt;% 
  group_by(gene_id) %&gt;%
  dplyr::count(gene_id) %&gt;% 
  as.data.frame() %&gt;% 
  set_colnames(c(&quot;gene_id&quot;, &quot;tx_n&quot;)) %&gt;%
   left_join(genedf, by = &quot;gene_id&quot;)</code></pre>
<pre class="r"><code>PALETTE_TTYPE = c(
    &quot;Lung&quot;= &quot;#007fff&quot;,
    &quot;Prostate&quot;= &quot;#665d1e&quot;,
    &quot;Stomach&quot;= &quot;#ffbf00&quot;,
    &quot;Central Nervous System&quot;= &quot;#fbceb1&quot;,
    &quot;Skin&quot;= &quot;#ff033e&quot;,
    &quot;Bladder&quot;= &quot;#ab274f&quot;,
    &quot;Haematopoietic and Lymphoid&quot;= &quot;#d5e6f7&quot;,
    &quot;Kidney&quot;= &quot;#7cb9e8&quot;,
    &quot;Thyroid&quot;= &quot;#efdecd&quot;,
    &quot;Soft Tissue&quot;= &quot;#8db600&quot;,
    &quot;Head and Neck&quot;= &quot;#e9d66b&quot;,
    &quot;Ovary&quot;= &quot;#b284be&quot;,
    &quot;Bone&quot;= &quot;#b2beb5&quot;,
    &quot;Endometrium&quot;= &quot;#10b36f&quot;,
    &quot;Breast&quot;= &quot;#6e7f80&quot;,
    &quot;Pancreas&quot;= &quot;#ff7e00&quot;,
    &quot;Peripheral Nervous System&quot;= &quot;#87a96b&quot;,
    &quot;Cervix&quot;= &quot;#c9ffe5&quot;,
    &quot;Large Intestine&quot;= &quot;#9f2b68&quot;,
    &quot;Liver&quot;= &quot;#00ffff&quot;,
    &quot;Vulva&quot;= &quot;#008000&quot;,
    &quot;Esophagus&quot;= &quot;#cd9575&quot;,
    &quot;Biliary Tract&quot;= &quot;#72a0c1&quot;,
    &quot;Other tissue&quot;= &quot;#a32638&quot;,
    &quot;Small Intestine&quot;= &quot;#9966cc&quot;,
    &quot;Placenta&quot;= &quot;#f19cbb&quot;,
    &quot;Testis&quot;= &quot;#e32636&quot;,
    &quot;Adrenal Gland&quot;= &quot;#3b7a57&quot;,
    &quot;Other&quot;= &quot;whitesmoke&quot;)</code></pre>
</div>
</div>
<div id="figure-1" class="section level1">
<h1>Figure 1</h1>
<p>mRNA-protein correlations across tissues and cancer related
pathways.</p>
<div id="figure-1a" class="section level2">
<h2>Figure 1A</h2>
<pre class="r"><code>correlation_data_tissue = read.csv(&quot;output/correlations-datasets/correlation-tissue-nontransformed.csv&quot;, row.names = 1) </code></pre>
<pre class="r"><code>sample_md = read.delim(&quot;~/Documents/2023/RNA-protein-correlationx/Data/e0022_diann_051021_sample_mapping_averaged.txt&quot;)</code></pre>
<pre class="r"><code>matrix_tissue = pivot_wider(correlation_data_tissue, names_from  = Tissue, values_from = Correlation)  %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;Gene&quot;) 

tissue_count =  sample_md %&gt;% group_by(Tissue_type) %&gt;% summarise(Count = n()) %&gt;% as.data.frame() %&gt;% 
  dplyr::rename(Tissue = Tissue_type)

medians_tissue = aggregate(Correlation ~ Tissue, data = correlation_data_tissue, FUN = median) %&gt;% 
  as.data.frame() %&gt;%
   left_join(tissue_count, by = &quot;Tissue&quot;) %&gt;%
    mutate(Tissue = recode(Tissue, 
                           &quot;Breast&quot; = &quot;Breast&quot;,
                           &quot;Central Nervous System&quot; = &quot;Central Nervous\nSystem&quot;,
                           &quot;Head and Neck&quot; = &quot;Head and\nNeck&quot;,
                           &quot;Esophagus&quot; = &quot;Esophagus&quot;, 
                         &quot;Bladder&quot; = &quot;Bladder&quot;, 
                         &quot;Skin&quot; = &quot;Skin&quot;, 
                         &quot;Large Intestine&quot; = &quot;Large Intestine\n&quot;, 
                         &quot;Kidney&quot; = &quot;Kidney&quot;, 
                         &quot;Pancreas&quot; = &quot;Pancreas&quot;, 
                         &quot;Soft Tissue&quot; = &quot;Soft Tissue&quot;, 
                         &quot;Thyroid&quot; = &quot;Thyroid&quot;, 
                         &quot;Peripheral Nervous System&quot; = &quot;Peripheral Nervous\nSystem&quot;, 
                         &quot;Bone&quot; = &quot;Bone&quot;, 
                         &quot;Ovary&quot; = &quot;Ovary&quot;, 
                         &quot;Haematopoietic and Lymphoid&quot; = &quot;Haematopoietic and\nLymphoid&quot;, 
                         &quot;Stomach&quot; = &quot;Stomach&quot;, 
                         &quot;Lung&quot; = &quot;Lung&quot;, 
                         &quot;Liver&quot; = &quot;Liver&quot;, 
                         &quot;Cervix&quot; = &quot;Cervix&quot;)) %&gt;%
     mutate(Label = paste0(Tissue, &quot; (n=&quot;, Count, &quot;)&quot;)) %&gt;%
  arrange(Correlation) 



tissue_dists= correlation_data_tissue %&gt;% 
  left_join(tissue_count, by = &quot;Tissue&quot;) %&gt;%
    group_by(Tissue) %&gt;% 
  mutate(Label = recode(Tissue, 
                           &quot;Breast&quot; = &quot;Breast&quot;,
                           &quot;Central Nervous System&quot; = &quot;Central Nervous\nSystem&quot;,
                           &quot;Head and Neck&quot; = &quot;Head and\nNeck&quot;,
                           &quot;Esophagus&quot; = &quot;Esophagus&quot;, 
                         &quot;Bladder&quot; = &quot;Bladder&quot;, 
                         &quot;Skin&quot; = &quot;Skin&quot;, 
                         &quot;Large Intestine&quot; = &quot;Large Intestine\n&quot;, 
                         &quot;Kidney&quot; = &quot;Kidney&quot;, 
                         &quot;Pancreas&quot; = &quot;Pancreas&quot;, 
                         &quot;Soft Tissue&quot; = &quot;Soft Tissue&quot;, 
                         &quot;Thyroid&quot; = &quot;Thyroid&quot;, 
                         &quot;Peripheral Nervous System&quot; = &quot;Peripheral Nervous\nSystem&quot;, 
                         &quot;Bone&quot; = &quot;Bone&quot;, 
                         &quot;Ovary&quot; = &quot;Ovary&quot;, 
                         &quot;Haematopoietic and Lymphoid&quot; = &quot;Haematopoietic and\nLymphoid&quot;, 
                         &quot;Stomach&quot; = &quot;Stomach&quot;, 
                         &quot;Lung&quot; = &quot;Lung&quot;, 
                         &quot;Liver&quot; = &quot;Liver&quot;, 
                         &quot;Cervix&quot; = &quot;Cervix&quot;)) %&gt;%
   mutate(Label = paste0(Label, &quot; (n=&quot;, Count, &quot;)&quot;)) %&gt;%
  mutate(Label = factor(Label, levels = medians_tissue$Label)) %&gt;%
  ggplot(aes(x = fct_rev(Label), y = Correlation)) + 
  ggdist::stat_halfeye(
    aes(color = Tissue,
        fill = after_scale(lighten(color, .5))),
    adjust = .5, 
    width = .75, 
    .width = 0,
    justification = -.4, 
    point_color = NA) + theme(legend.position = &quot;none&quot;) +
  geom_boxplot(
    aes(color = Tissue,
        color = after_scale(darken(color, .1, space = &quot;HLS&quot;))),
    width = .42, 
    outlier.shape = NA
  )  + 
  stat_summary(
    geom = &quot;text&quot;,
    fun = &quot;median&quot;,
    aes(label = round(..y.., 2),
        color = Tissue,
        color = after_scale(darken(color, .1, space = &quot;HLS&quot;))),
    fontface = &quot;bold&quot;,
    size = 8,
    vjust = 10.5,
    na.rm = TRUE
  ) +
  scale_color_manual(values = PALETTE_TTYPE, guide = &quot;none&quot;) +
  scale_fill_manual(values = PALETTE_TTYPE, guide = &quot;none&quot;) +
  theme_bw()  + 
  labs(y=&quot;mRNA-protein correlation&quot;,
       x= &quot;&quot;) +
  theme(axis.text.x = element_text(angle=70,  hjust = 1, size =30),
        axis.text.y = element_text(size =30),
    axis.text = element_text(color = &quot;black&quot;, 
                                 family=&quot;serif&quot;, 
                                 size=32), 
    axis.title = element_text(color = &quot;black&quot;, 
                                 family=&quot;serif&quot;, 
                                 size=32)) 

tissue_dists</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-6-1.png" alt="Distribution of mRNA-protein correlations across tissues in the ProCan-DepMapSanger dataset. Only tissues with at least 10 cell lines are shown." width="1920" />
<p class="caption">
Distribution of mRNA-protein correlations across tissues in the
ProCan-DepMapSanger dataset. Only tissues with at least 10 cell lines
are shown.
</p>
</div>
<pre class="r"><code> ggsave(filename = &quot;output/Publication/Figures/Fig_1A.png&quot;, height =10, width =25, plot = tissue_dists, dpi = 1200)</code></pre>
</div>
<div id="figure-1b-and-figure-1c" class="section level2">
<h2>Figure 1B and Figure 1C</h2>
<pre class="r"><code>cptac = read.csv(&quot;output/correlations-datasets/correlation-overall-cptac-linkedomics.csv&quot;, header=TRUE)
ccle = read.csv(&quot;output/correlations-datasets/CCLE-correlations-overall.csv&quot;, header=TRUE)
correlation_overall= read.csv(&quot;output/correlations-datasets/correlation-all-nontransformed.csv&quot;,
                            header=TRUE, row.names = 1)</code></pre>
<pre class="r"><code>cptac %&gt;% 
  mutate(Protein =  gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, Protein)) %&gt;% 
   mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein,
                        keytype = &quot;ENSEMBL&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;% 
  dplyr::rename(&quot;CPTAC&quot;= &quot;Correlation&quot;) %&gt;%
  inner_join(correlation_overall %&gt;% 
               dplyr::rename(&quot;Gene&quot; = &quot;Protein&quot;), by = &quot;Gene&quot;) %&gt;%
  with(cor(.$Correlation, .$CPTAC, method = &quot;spearman&quot;))</code></pre>
<pre><code>[1] 0.4315774</code></pre>
<pre class="r"><code>cptac %&gt;% 
  mutate(Protein =  gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, Protein)) %&gt;% 
   mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein,
                        keytype = &quot;ENSEMBL&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;% 
  dplyr::rename(&quot;CPTAC&quot;= &quot;Correlation&quot;) %&gt;%
  inner_join(ccle %&gt;% 
               dplyr::rename(&quot;Gene&quot; = &quot;Protein&quot;), by = &quot;Gene&quot;) %&gt;%
  with(cor(.$Correlation, .$CPTAC, method = &quot;spearman&quot;))</code></pre>
<pre><code>[1] 0.3019059</code></pre>
<pre class="r"><code>ccle %&gt;%
  dplyr::rename(&quot;CCLE&quot;= &quot;Correlation&quot;) %&gt;%
  inner_join(correlation_overall,
               by = &quot;Protein&quot;) %&gt;%
  with(cor(.$Correlation, .$CCLE, method = &quot;spearman&quot;))</code></pre>
<pre><code>[1] 0.3846969</code></pre>
<pre class="r"><code>overall_correlations_cell_lines = correlation_overall  %&gt;%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.1, y = 600,
  label = &quot;paste(italic(median), \&quot; = 0.42\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept = 0.42, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;ProCan-DepMapSanger&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) </code></pre>
<pre class="r"><code>overall_correlations_ccle = ccle   %&gt;%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.1, y = 700,
  label = &quot;paste(italic(median), \&quot; = 0.46\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept = 0.4555411, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;CCLE&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) 


#median(cptac$Correlation,na.rm = TRUE)
overall_correlations_cptac = cptac  %&gt;%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color=&quot;white&quot;, alpha=1, fill = &quot;#ccd5ae&quot;) + theme_bw() +
  labs(x=&quot;mRNA-protein correlation&quot;, 
       y = &quot;Number of associations&quot;) +  annotate(&quot;text&quot;, x = -0.2, y = 1250,
  label = &quot;paste(italic(median), \&quot; = 0.38\&quot;)&quot;, parse = TRUE, color = &quot;#9a031e&quot;) + 
  geom_vline(xintercept =0.3816182, linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;) +
  ggtitle(&quot;CPTAC&quot;) +
     theme(legend.position = &quot;none&quot;,
      axis.text=element_text(size=15,  family=&quot;serif&quot;, color = &quot;black&quot;),
      axis.title = element_text(size=16, family = &quot;serif&quot;, color = &quot;black&quot;)) 


ggarrange(overall_correlations_cell_lines,overall_correlations_cptac,overall_correlations_ccle, ncol= 3)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-12-1.png" alt="Histogram of mRNA-protein correlations across each cohort for ProCan-DepMapSanger, CPTAC and CCLE with medians indicated" width="960" />
<p class="caption">
Histogram of mRNA-protein correlations across each cohort for
ProCan-DepMapSanger, CPTAC and CCLE with medians indicated
</p>
</div>
<pre class="r"><code> ggsave(filename = &quot;output/Publication/Figures/Fig_1B.png&quot;, height =4, width =4, plot = overall_correlations_cell_lines, dpi = 1200)
 
 
ggsave(filename = &quot;output/Publication/Figures/Fig_1C_left.png&quot;, height =4, width =4, plot = overall_correlations_cptac, dpi = 1200)
 
ggsave(filename = &quot;output/Publication/Figures/Fig_1C_right.png&quot;, height =4, width =4, plot = overall_correlations_ccle, dpi = 1200)</code></pre>
</div>
<div id="figure-1d" class="section level2">
<h2>Figure 1D</h2>
<p>Selected pathways were retrieved from Arad &amp; Geiger (2023).</p>
<pre class="r"><code>kegg_pathways = msigdbr(species=&quot;human&quot;, category = &quot;C2&quot;, subcategory = &quot;CP:KEGG&quot;) %&gt;%
  as.data.frame() %&gt;%
  dplyr::filter(gs_description %in% c(&quot;Cell cycle&quot;, &quot;Oxidative phosphorylation&quot;,
                                      &quot;Ribosome&quot;, &quot;Proteasome&quot;, &quot;Spliceosome&quot;,
                                      &quot;Focal adhesion&quot;, &quot;Apoptosis&quot;) )

lists = split(kegg_pathways, kegg_pathways$gs_description)

kegg_genes = lapply(lists , function(x){
  x %&lt;&gt;% dplyr::select(Gene = gene_symbol)
})

pathway_genes = read_excel_allsheets(&quot;data/Pathways/Pathways-member-genes.xlsx&quot;)
pathway_genes = lapply(pathway_genes[1:10] , function(x){
  x %&lt;&gt;% dplyr::select(Gene)
  }
                       )


tissue_matrix = correlation_data_tissue %&gt;%
  pivot_wider(names_from  = Tissue, values_from = Correlation)  %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(-Protein)



correlation_per_kegg = lapply(names(kegg_genes), function(x) {
  pathways = kegg_genes[[x]]$Gene
  pathway_medians = tissue_matrix %&gt;%
  dplyr::filter(Gene %in% pathways) %&gt;%
  dplyr::select(-c(Gene)) %&gt;%
  summarise_all(.funs = ~median(., na.rm = TRUE)) %&gt;%
  mutate(Pathway = x)

  return(pathway_medians)


})

correlation_per_kegg %&lt;&gt;% do.call(rbind,.)




correlation_per_pathway = lapply(names(pathway_genes), function(x) {
  pathways = pathway_genes[[x]]$Gene
  pathway_medians = tissue_matrix %&gt;%
  dplyr::filter(Gene %in% pathways) %&gt;%
  dplyr::select(-c(Gene)) %&gt;%
  summarise_all(.funs = ~median(., na.rm = TRUE)) %&gt;%
  mutate(Pathway = x)

  return(pathway_medians)


})</code></pre>
<pre class="r"><code>plot = do.call(rbind, correlation_per_pathway) %&gt;%
  dplyr::filter(Pathway %in% c(&quot;PI3K&quot;, &quot;NOTCH&quot;, &quot;WNT&quot;, &quot;RTK RAS&quot;, &quot;TGF-Beta&quot;,
                               &quot;HIPPO&quot;)) %&gt;%
  rbind(correlation_per_kegg) %&gt;%
  mutate(Pathway = ifelse(Pathway == &quot;Oxidative phosphorylation&quot;, &quot;OXPHOS&quot;, Pathway)) %&gt;%
  melt() %&gt;%
  mutate(Pathway = factor(Pathway, levels =
                            c(&quot;Ribosome&quot;, &quot;OXPHOS&quot;,
                              &quot;Proteasome&quot;, &quot;Spliceosome&quot;, &quot;PI3K&quot;,
                              &quot;NOTCH&quot;, &quot;Cell cycle&quot;, &quot;Apoptosis&quot;,
                              &quot;WNT&quot;, &quot;RTK RAS&quot;, &quot;TGF-Beta&quot;, &quot;HIPPO&quot;,
                              &quot;Focal adhesion&quot;))) %&gt;%
  ggplot(aes(x=Pathway, y=value)) + geom_boxplot() +
  theme_bw() +   theme_bw() +  #geom_jitter(aes(color = variable), size=4, shape = 1, stroke = 1.5) +
  geom_jitter(
    aes(color = variable,
        #color = after_scale(darken(color, .1, space = &quot;HLS&quot;)),
        color = after_scale(desaturate(lighten(color, .25), .4))),
    size=3, stroke = 1
  ) +
 scale_color_manual(
  values = PALETTE_TTYPE, 
  labels = c(&quot;Breast&quot; = &quot;Breast&quot;,
                           &quot;Central Nervous System&quot; = &quot;Central Nervous\nSystem&quot;,
                           &quot;Head and Neck&quot; = &quot;Head and\nNeck&quot;,
                           &quot;Esophagus&quot; = &quot;Esophagus&quot;, 
                         &quot;Bladder&quot; = &quot;Bladder&quot;, 
                         &quot;Skin&quot; = &quot;Skin&quot;, 
                         &quot;Large Intestine&quot; = &quot;Large Intestine&quot;, 
                         &quot;Kidney&quot; = &quot;Kidney&quot;, 
                         &quot;Pancreas&quot; = &quot;Pancreas&quot;, 
                         &quot;Soft Tissue&quot; = &quot;Soft Tissue&quot;, 
                         &quot;Thyroid&quot; = &quot;Thyroid&quot;, 
                         &quot;Peripheral Nervous System&quot; = &quot;Peripheral Nervous\nSystem&quot;, 
                         &quot;Bone&quot; = &quot;Bone&quot;, 
                         &quot;Ovary&quot; = &quot;Ovary&quot;, 
                         &quot;Haematopoietic and Lymphoid&quot; = &quot;Haematopoietic\nand Lymphoid&quot;, 
                         &quot;Stomach&quot; = &quot;Stomach&quot;, 
                         &quot;Lung&quot; = &quot;Lung&quot;, 
                         &quot;Liver&quot; = &quot;Liver&quot;, 
                         &quot;Cervix&quot; = &quot;Cervix&quot;))  + # Split into 6 rows
  labs(y= &quot;Median mRNA-protein correlation&quot;,
       color = &quot;Tissue&quot;) + theme(legend.position = &quot;top&quot;,
        legend.spacing.y = unit(0, &quot;mm&quot;),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = &quot;black&quot;),
    legend.text = element_text(family = &quot;serif&quot;, size=18),
    legend.title = element_text(family = &quot;serif&quot;, size=18)) +
  theme(axis.text.x = element_text( color = &quot;black&quot;, size =25, family = &quot;serif&quot;, hjust = 1),
        axis.text.y = element_text(color = &quot;black&quot;, size =25, family = &quot;serif&quot;),
        axis.title = element_text(color = &quot;black&quot;, size =25, family = &quot;serif&quot;),
        legend.position = &quot;right&quot;, legend.spacing.x = unit(0.8, &#39;cm&#39;)) + xlab(&quot;&quot;) + coord_flip()
plot</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-15-1.png" alt="Boxplot showing median mRNA-protein correlations by tissue in the ProCan-DepMapSanger dataset for key biological and selected cancer-related pathways" width="1056" />
<p class="caption">
Boxplot showing median mRNA-protein correlations by tissue in the
ProCan-DepMapSanger dataset for key biological and selected
cancer-related pathways
</p>
</div>
<pre class="r"><code>ggsave(filename = &quot;output/Publication/Figures/Fig_1D.png&quot;, height =11, width =11, plot = plot, dpi = 1200)</code></pre>
</div>
<div id="figure-1e" class="section level2">
<h2>Figure 1E</h2>
<pre class="r"><code>pathways_across_datasets = read.table(&quot;output/Other/Pathways_median_all_datasets.tsv&quot;, sep = &quot;\t&quot;, header =TRUE)



median_pathway &lt;- pathways_across_datasets %&gt;%
  group_by(Pathway, Dataset) %&gt;%  # Group by Pathway and Dataset
  summarise(Median_Correlation = median(`Median.Correlation`, na.rm = TRUE), .groups = &quot;drop&quot;) %&gt;% 
  as.data.frame() %&gt;% 
  pivot_wider(names_from = Dataset, values_from = Median_Correlation) %&gt;% as.data.frame() %&gt;% 
  dplyr::rename(ProCan = `ProCan-DepMapSanger`)

library(ggpubr)



pathways_cor_1 = median_pathway %&gt;%

 ggplot(aes(x = `CPTAC`, y = ProCan)) +
  geom_point(size =3) + geom_smooth(method = &quot;lm&quot;, color = &quot;blue&quot;, fill = &quot;lightgray&quot;, se = TRUE) +
   theme_bw() + xlab(&quot;CPTAC&quot;) +
  ylab(&quot;ProCan-DepMapSanger&quot;) +
  annotate(&quot;text&quot;, x = 0.22, y = 0.5, size = 5,
  label = &quot;paste(italic(r), \&quot; = 0.85, \&quot;, italic(p), \&quot; = 0.00034\&quot;)&quot;, 
  parse = TRUE, color = &quot;black&quot;) +
  theme(axis.text.x = element_text(size = 25, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 25, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=25, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;,
        strip.text.y = element_text(
        size = 16, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 16, family = &quot;serif&quot;
        )) 
 

 
ggsave(filename = &quot;output/Publication/Figures/Fig_1E_upper.png&quot;,height = 5, width = 8, plot =pathways_cor_1, dpi = 1200 )


pathways_cor_2 = median_pathway %&gt;% 
 ggplot(aes(x = `CPTAC`, y = CCLE)) +
  geom_point(size =3) + geom_smooth(method = &quot;lm&quot;, color = &quot;blue&quot;, fill = &quot;lightgray&quot;, se = TRUE) +
   theme_bw() + xlab(&quot;CPTAC&quot;) +
  ylab(&quot;CCLE&quot;) +
  annotate(&quot;text&quot;, x = 0.22, y = 0.57, size = 5,
  label = &quot;paste(italic(r), \&quot; = 0.74, \&quot;, italic(p), \&quot; = 0.0053\&quot;)&quot;, 
  parse = TRUE, color = &quot;black&quot;) +
  theme(axis.text.x = element_text(size = 25, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 25, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=25, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;,
        strip.text.y = element_text(
        size = 16, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 16, family = &quot;serif&quot;
        )) 
 

 ggsave(filename = &quot;output/Publication/Figures/Fig_1E_lower.png&quot;,height = 5, width = 8, plot =pathways_cor_2, dpi = 1200 )

 
 ggarrange(pathways_cor_1, pathways_cor_2, nrow=2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-16-1.png" alt="Correlation of median mRNA-protein correlations per tissue for CPTAC with ProCan-DepMapSanger (top), and with CCLE (bottom), for pathways in Figure 1d" width="480" />
<p class="caption">
Correlation of median mRNA-protein correlations per tissue for CPTAC
with ProCan-DepMapSanger (top), and with CCLE (bottom), for pathways in
Figure 1d
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/0462df41c2a09e5ded2a36493174bfe6c2de2586/docs/figure/publication-figures.Rmd/unnamed-chunk-16-1.png" target="_blank">0462df4</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="figure-2" class="section level1">
<h1>Figure 2</h1>
<p>mRNA-protein correlations of cancer driver genes from OncoKB.</p>
<div id="figure-2a" class="section level2">
<h2>Figure 2A</h2>
<pre class="r"><code>drivers = read.delim(here::here(&quot;data/OncoKb/cancerGeneList.tsv&quot;), fill=TRUE, check.names = FALSE)</code></pre>
<pre class="r"><code>correlation_data_tissue = read.csv(&quot;output/correlations-datasets/correlation-tissue-nontransformed.csv&quot;, row.names = 1) 
matrix_tissue = pivot_wider(correlation_data_tissue, names_from  = Tissue, values_from = Correlation)  %&gt;% 
  as.data.frame() %&gt;% 
  column_to_rownames(&quot;Gene&quot;) 


driver_matrix = matrix_tissue %&gt;%
  rownames_to_column(&quot;Gene&quot;) %&gt;%
  dplyr::filter(Gene %in% drivers$`Hugo Symbol`) %&gt;%
  dplyr::select(-Protein) %&gt;%
  column_to_rownames(&quot;Gene&quot;)


tmp=rowSums(is.na(driver_matrix) )

tmp=tmp[tmp&lt;5]


matrix_drivers=driver_matrix[names(tmp),]

matrix_drivers %&lt;&gt;% as.matrix()


#km = kmeans(na.omit(matrix_drivers), 10)</code></pre>
<pre class="r"><code>heatmap_viz= matrix_drivers %&gt;%
  #  drop_na() %&gt;%

pheatmap::pheatmap( border_color = &quot;white&quot;,
         breaks = seq(-1,1, length.out=100),
         color = colorRampPalette(c(&quot;#2F124b&quot;, &quot;#6e74b4&quot;, &quot;#f9f9f9&quot;,
                                    &quot;#fd9675&quot;, &quot;#Ea2a5f&quot;))(100),
         na_col = &quot;darkgrey&quot;, cutree_rows    = 5, show_rownames = FALSE, fontsize_col=12)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-19-1.png" alt=" Heatmap of mRNA-protein correlations across tissue types for cancer driver genes from OncoKB. Clusters identidied by hierachical clustering using Euclidean distance." width="576" />
<p class="caption">
Heatmap of mRNA-protein correlations across tissue types for cancer
driver genes from OncoKB. Clusters identidied by hierachical clustering
using Euclidean distance.
</p>
</div>
<pre class="r"><code>ggsave(plot = heatmap_viz, filename = &quot;output/Publication/Figures/Fig_2A.png&quot;,
      height = 8, width = 8, dpi = 1200)</code></pre>
</div>
<div id="figure-2b" class="section level2">
<h2>Figure 2B</h2>
<pre class="r"><code>matrix_tissue = pivot_wider(correlation_data_tissue, names_from  = Tissue, values_from = Correlation)  %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;Gene&quot;) %&gt;%
  dplyr::select(-Protein)</code></pre>
<pre class="r"><code>drivers = read.delim(here::here(&quot;data/OncoKb/cancerGeneList.tsv&quot;), fill=TRUE, check.names = FALSE)
tmp=rowSums(is.na(matrix_tissue ) )

tmp=tmp[tmp&lt;5]

matrix_filtered=matrix_tissue[names(tmp),]

n_counts &lt;- correlation_overall %&gt;%
  dplyr::filter(Protein %in% rownames(matrix_filtered)) %&gt;%
  mutate(Gene = ifelse(Protein %in% drivers$`Hugo Symbol`, &quot;Cancer Driver&quot;, &quot;Non-Driver&quot;)) %&gt;%
  ungroup() %&gt;%  # Remove any grouping
  mutate(Gene = as.character(Gene)) %&gt;%  # Convert to character
  group_by(Gene) %&gt;%  # Group by Gene
  summarise(n = n(), .groups = &quot;drop&quot;) %&gt;%  # Count occurrences
  mutate(Label = paste0(Gene, &quot;\n(n = &quot;, n, &quot;)&quot;))  # Create formatted labels


drivers_v_nondrivers &lt;- correlation_overall %&gt;%
  dplyr::filter(Protein %in% rownames(matrix_filtered)) %&gt;%
  mutate(Gene = ifelse(Protein %in% drivers$`Hugo Symbol`, &quot;Cancer Driver&quot;, &quot;Non-Driver&quot;)) %&gt;%
  mutate(Gene = factor(Gene, levels = c(&quot;Non-Driver&quot;, &quot;Cancer Driver&quot;))) %&gt;%
  ggplot(aes(x = Gene, y = Correlation, fill = Gene)) +

  # Violin plot
  geom_violin(alpha = 0.6) +

  # Boxplot
  geom_boxplot(width = 0.2, fill = &quot;white&quot;, outlier.shape = NA) +

  # Add significance test
  geom_signif(comparisons = list(c(&quot;Non-Driver&quot;, &quot;Cancer Driver&quot;)),
              map_signif_level = TRUE,
               annotations = &quot;****&quot;) +



  # Custom colors
  scale_fill_manual(values = c(&quot;Cancer Driver&quot; = &quot;#f08080&quot;, &quot;Non-Driver&quot; = &quot;#adb5bd&quot;)) +
    scale_x_discrete(labels = setNames(n_counts$Label, n_counts$Gene)) +

  # Labels and theme
  labs(y = &quot;mRNA-protein correlation&quot;, x = &quot;&quot;) +
  theme_bw() +
  theme(
    axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
    axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
    axis.title = element_text(size = 20, family = &quot;serif&quot;),
    legend.box.background = element_rect(color = &quot;black&quot;),
    legend.text = element_text(family = &quot;serif&quot;, size=15),
    legend.title = element_text(family = &quot;serif&quot;, size=15),
    plot.title = element_text(family = &quot;serif&quot;, size = 20),
    panel.grid.major.x = element_blank(),
    panel.grid.major.y = element_line(size = 0.1),
    legend.position = &quot;top&quot;,
    strip.text.y = element_text(size = 10, face = &quot;bold.italic&quot;, family = &quot;serif&quot;),
    strip.text.x = element_text(size = 10, family = &quot;serif&quot;)
  )

drivers_v_nondrivers</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-21-1.png" alt="Violin plot showing mRNA-protein correlations of cancer driver and non-driver genes with median and standard deviation indicated. **** indicates P &lt; 0.0001 by unpaired Student’s t-test." width="480" />
<p class="caption">
Violin plot showing mRNA-protein correlations of cancer driver and
non-driver genes with median and standard deviation indicated. ****
indicates P &lt; 0.0001 by unpaired Student’s t-test.
</p>
</div>
<pre class="r"><code>ggsave(filename = &quot;output/Publication/Figures/Fig_2B.png&quot;,height = 6.5, width = 6, plot =drivers_v_nondrivers, dpi = 1200 )</code></pre>
</div>
<div id="figure-2c" class="section level2">
<h2>Figure 2C</h2>
<pre class="r"><code>matrix &lt;- matrix_drivers[c(heatmap_viz$tree_row[[&quot;order&quot;]]),heatmap_viz$tree_col[[&quot;order&quot;]]]
clusters= cutree(heatmap_viz$tree_row, k=5)[heatmap_viz$tree_row[[&quot;order&quot;]]] %&gt;%
  as.data.frame() %&gt;%
  set_colnames(&quot;Cluster&quot;)


clusters$clusters &lt;- NA

clusters$clusters[clusters$Cluster == 5] &lt;- 1
clusters$clusters[clusters$Cluster == 1] &lt;- 2
clusters$clusters[clusters$Cluster == 4] &lt;- 3
clusters$clusters[clusters$Cluster == 3] &lt;- 4
clusters$clusters[clusters$Cluster == 2] &lt;- 5

# 
# clusters %&gt;%
#    as.data.frame() %&gt;%
#    rownames_to_column(&quot;Gene&quot;) %&gt;%
#    inner_join(matrix_drivers %&gt;% melt() %&gt;%
#                 dplyr::rename(&quot;Gene&quot; = &quot;Var1&quot;,  &quot;Correlation&quot; = &quot;value&quot; ), by = &quot;Gene&quot;) %&gt;%
#    group_by(Cluster) %&gt;%
#    mutate(median= median(Correlation, na.rm =TRUE)) %&gt;%
#    distinct(&quot;median&quot;, .keep_all = T)


clusters_drivers = clusters %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Gene&quot;) %&gt;%
  inner_join(matrix_drivers %&gt;% melt() %&gt;%
               dplyr::rename(&quot;Gene&quot; = &quot;Var1&quot;,  &quot;Correlation&quot; = &quot;value&quot; ), by = &quot;Gene&quot;) %&gt;%
  group_by(clusters) %&gt;%
  mutate(median= median(Correlation, na.rm =TRUE)) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=Correlation)) +
   geom_histogram(color=&quot;black&quot;, alpha=0.4, fill = &quot;#6e74b4&quot;) + theme_bw() +
  facet_grid(rows = vars(Cluster), scales = &quot;free_y&quot;) + ylab(&quot;Number of Proteins&quot;) +
  geom_vline(aes(xintercept = median), linetype = &quot;dashed&quot;, color = &quot;#9a031e&quot;, size =1) +
    theme(legend.position = &quot;top&quot;,
                          legend.title = element_blank(),
        legend.spacing.y = unit(0, &quot;mm&quot;),
        panel.border = element_rect(colour = &quot;black&quot;, fill=NA),
        legend.background = element_blank(),
        legend.box.background = element_rect(colour = &quot;black&quot;)) +
  theme(axis.text.x = element_text( color = &quot;black&quot;, size =16, family = &quot;serif&quot;, hjust = 1),
        axis.text.y = element_text(color = &quot;black&quot;, size =16, family = &quot;serif&quot;),
        axis.title = element_text(color = &quot;black&quot;, size =16, family = &quot;serif&quot;),
        legend.position = &quot;top&quot;) + xlab(&quot;mRNA-protein correlation&quot;)

clusters_drivers</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-22-1.png" alt="Distribution of mRNA-protein correlations for genes in each cluster identified in heatmap in Figure 2A" width="672" />
<p class="caption">
Distribution of mRNA-protein correlations for genes in each cluster
identified in heatmap in Figure 2A
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/0462df41c2a09e5ded2a36493174bfe6c2de2586/docs/figure/publication-figures.Rmd/unnamed-chunk-22-1.png" target="_blank">0462df4</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code> # ggsave(plot = clusters_drivers, filename = &quot;output/Publication/Figures/Figure_2b_update.png&quot;,
#        height = 8, width = 7, dpi = 1200)

# 
 cluster_full = correlation_data_tissue %&gt;%
   left_join(clusters %&gt;%
               rownames_to_column(&quot;Gene&quot;), by = &quot;Gene&quot;)
write.csv(cluster_full, &quot;output/Publication/Data/clusters.csv&quot;)</code></pre>
<div id="gene-ontologies" class="section level3">
<h3>Gene ontologies</h3>
<p>GO terms for each cluster were identified using the clusterProfiler
and saved in a CSV. The top terms were combined with the distribution
plot in Adobe Illustrator.</p>
<pre class="r"><code>universe = correlation_data_tissue$Protein %&gt;%
  as.data.frame() %&gt;%
  set_colnames(&quot;Proteins&quot;) %&gt;%
  mutate(&quot;Protein_name&quot; = gsub(&quot;;.*&quot;, &quot;&quot;, Proteins)) %&gt;%
  mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = &quot;UNIPROT&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;%
  mutate(&quot;Entrez&quot; = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = &quot;UNIPROT&quot;, column = &quot;ENTREZID&quot;, multiVals=&quot;first&quot;))</code></pre>
<pre class="r"><code>library(clusterProfiler)
gene_ontology = list()
for (clus in unique(clusters$clusters)) {
  df =  clusters %&gt;%
    dplyr::filter(clusters == clus) %&gt;%
    rownames_to_column(&quot;Gene&quot;) %&gt;%
      mutate(&quot;Entrez&quot; = mapIds(org.Hs.eg.db, keys = Gene,
                         keytype = &quot;SYMBOL&quot;, column = &quot;ENTREZID&quot;, multiVals=&quot;first&quot;))

  ego = enrichGO(gene          = df$Entrez,
                universe      = universe$Entrez,
                OrgDb         = org.Hs.eg.db,
                ont           = &quot;ALL&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
        readable      = TRUE)

  gene_ontology[[paste0(&quot;cluster&quot;, clus)]] = head(ego, n=100) %&gt;%
    as.data.frame() %&gt;%
    dplyr::select( Description, &quot;p.adjust&quot;, &quot;qvalue&quot;, geneID, Count) %&gt;%
    mutate(Cluster = paste(&quot;Cluster&quot;, clus))

}


gene_ontology_top = lapply(gene_ontology, function(x){
  ont= x %&gt;%
    arrange(qvalue) %&gt;%
    head(200)
  return(ont)
})

library(openxlsx)
write.xlsx(gene_ontology_top, file = &quot;output/Publication/Data/Gene-ontology-Figure2b.xlsx&quot;)</code></pre>
<pre class="r"><code>gene_ontology_top$cluster1 %&gt;% 
  DT::datatable(caption = &quot;Gene ontology for Cluster 1&quot;)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a9f1a161ba4406b13499" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a9f1a161ba4406b13499">{"x":{"filter":"none","vertical":false,"caption":"<caption>Gene ontology for Cluster 1<\/caption>","data":[["GO:0008134","GO:0061629"],["transcription factor binding","RNA polymerase II-specific DNA-binding transcription factor binding"],[0.008830567367185654,0.008830567367185654],[0.007170686283128201,0.007170686283128201],["CREB1/SMARCB1/PTPN2/SPEN/ELOC","CREB1/SMARCB1/PTPN2/SPEN"],[5,4],["Cluster 1","Cluster 1"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Description<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Description","targets":1},{"name":"p.adjust","targets":2},{"name":"qvalue","targets":3},{"name":"geneID","targets":4},{"name":"Count","targets":5},{"name":"Cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>gene_ontology_top$cluster2 %&gt;% 
  DT::datatable(caption = &quot;Gene ontology for Cluster 2&quot;)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-4f484109487b50d5022c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4f484109487b50d5022c">{"x":{"filter":"none","vertical":false,"caption":"<caption>Gene ontology for Cluster 2<\/caption>","data":[["GO:0032993","GO:0000785","GO:0140535","GO:0006259"],["protein-DNA complex","chromatin","intracellular protein-containing complex","DNA metabolic process"],[1.637332259588627e-05,2.852800239255602e-05,0.0003664755043557735,0.005437580824993429],[1.417464228665019e-05,2.469713930688569e-05,0.0003172635945234439,0.004836821568991546],["CTNNB1/UBE2A/SMARCE1/CTR9/SET/NCOR1/SF3B1/POLD1/EP400/FANCI/EZH2/PPP4R2/KAT7/EIF3E","CTNNB1/UBE2A/SMARCE1/CTR9/SET/NCOR1/SF3B1/EP400/FANCI/EZH2/PPP4R2/KAT7/EIF3E","CTNNB1/UBE2A/RNASEH2A/PSMB2/CTR9/CNOT3/MTOR/DCTN1/POLD1/EP400/POLG/KAT7","CTNNB1/UBE2A/SMARCE1/RNASEH2A/SET/PPP2R1A/POLD1/EP400/POLG/FANCI/EZH2/PPP4R2/KAT7"],[14,13,12,13],["Cluster 2","Cluster 2","Cluster 2","Cluster 2"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Description<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Description","targets":1},{"name":"p.adjust","targets":2},{"name":"qvalue","targets":3},{"name":"geneID","targets":4},{"name":"Count","targets":5},{"name":"Cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>gene_ontology_top$cluster3 %&gt;% 
  DT::datatable(caption = &quot;Gene ontology for Cluster 3&quot;)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-1d05f9b065ae358318d5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1d05f9b065ae358318d5">{"x":{"filter":"none","vertical":false,"caption":"<caption>Gene ontology for Cluster 3<\/caption>","data":[["GO:0005925","GO:0030055","GO:0090398","GO:0030670","GO:0030666","GO:0030863","GO:0030139","GO:0002474","GO:0045335","GO:0072359","GO:0071248","GO:0019885","GO:0035239","GO:0048002","GO:0010628","GO:0060341","GO:0002483","GO:0022603","GO:0050839","GO:0051015","GO:0042605","GO:0045296"],["focal adhesion","cell-substrate junction","cellular senescence","phagocytic vesicle membrane","endocytic vesicle membrane","cortical cytoskeleton","endocytic vesicle","antigen processing and presentation of peptide antigen via MHC class I","phagocytic vesicle","circulatory system development","cellular response to metal ion","antigen processing and presentation of endogenous peptide antigen via MHC class I","tube morphogenesis","antigen processing and presentation of peptide antigen","positive regulation of gene expression","regulation of cellular localization","antigen processing and presentation of endogenous peptide antigen","regulation of anatomical structure morphogenesis","cell adhesion molecule binding","actin filament binding","peptide antigen binding","cadherin binding"],[2.434738761253616e-07,2.434738761253616e-07,0.0002815877214049628,0.0005972885024064874,0.0005972885024064874,0.001138426385166563,0.001543491931933765,0.003432114348663572,0.004767437980017344,0.00466969896940829,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008792431743697426,0.008524933467522602,0.008524933467522602,0.008524933467522602,0.008524933467522602],[1.688344663578212e-07,1.688344663578212e-07,0.0002290596987578012,0.0004141835960813257,0.0004141835960813257,0.0007894301199209388,0.001070318675673701,0.002791879826594418,0.003305931051246749,0.003798602559392087,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007152271258530686,0.007412985623932698,0.007412985623932698,0.007412985623932698,0.007412985623932698],["RAC2/EGFR/RRAS2/IQGAP1/B2M/PTK7/CALR/TPM4/EZR/MSN/MYH9/RRAS/MDC1/HMGA1","RAC2/EGFR/RRAS2/IQGAP1/B2M/PTK7/CALR/TPM4/EZR/MSN/MYH9/RRAS/MDC1/HMGA1","PML/LMNA/B2M/CALR/HMGA2/HMGA1","RAC2/HLA-A/B2M/TAP1/CALR","RAC2/EGFR/HLA-A/B2M/TAP1/CALR","IQGAP1/TPM4/EZR/MYH9/NUMA1","RAC2/EGFR/HLA-A/PICALM/B2M/TAP1/CALR","HLA-A/B2M/TAP1/CALR","RAC2/HLA-A/B2M/TAP1/CALR","PML/EGFR/QKI/LMNA/HTATIP2/PTK7/CALR/MYH9/RRAS/GTF2I/HMGA2","EGFR/NQO1/IQGAP1/B2M/CALR","HLA-A/B2M/TAP1","PML/EGFR/QKI/HTATIP2/PTK7/MYH9/RRAS/GTF2I/HMGA2","HLA-A/B2M/TAP1/CALR","HLA-A/QKI/LMNA/B2M/CALR/EZR/MSN/MYH9/TP53BP1/AFDN/HMGA2","RAC2/PML/EGFR/PTPN1/LMNA/PICALM/EZR/MSN/NUMA1/BCL2L1/AFDN","HLA-A/B2M/TAP1","PML/HTATIP2/PTK7/EZR/MSN/MYH9/RRAS/GTF2I/HMGA2","EGFR/PTPN1/IQGAP1/PICALM/CALR/NDRG1/EZR/MSN/MYH9/AFDN","EGFR/IQGAP1/TPM4/EZR/MYH9/AFDN","HLA-A/B2M/TAP1","EGFR/PTPN1/IQGAP1/PICALM/NDRG1/EZR/MYH9/AFDN"],[14,14,6,5,6,5,7,4,5,11,5,3,9,4,11,11,3,9,10,6,3,8],["Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3","Cluster 3"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Description<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Description","targets":1},{"name":"p.adjust","targets":2},{"name":"qvalue","targets":3},{"name":"geneID","targets":4},{"name":"Count","targets":5},{"name":"Cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>gene_ontology_top$cluster4 %&gt;% 
  DT::datatable(caption = &quot;Gene ontology for Cluster 4&quot;)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-bed2a3df69eb49c3214f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bed2a3df69eb49c3214f">{"x":{"filter":"none","vertical":false,"caption":"<caption>Gene ontology for Cluster 4<\/caption>","data":[["GO:0003682","GO:0051276","GO:0098687","GO:0000819","GO:0033044","GO:0034605","GO:0034399","GO:0000775","GO:0098813","GO:0098978","GO:0070603","GO:0090307","GO:0007059","GO:1903047","GO:0009408","GO:0007062","GO:0030071","GO:1902099","GO:0007091","GO:0044784","GO:0007064","GO:0097431","GO:0051338","GO:0014070","GO:0033045","GO:0000278","GO:0000793","GO:0000785","GO:0016363","GO:0009266","GO:1904949","GO:0016586","GO:1902459","GO:0045595","GO:0071867","GO:0071868","GO:0071869","GO:0071870","GO:0071407","GO:0019827","GO:0051225","GO:0098727","GO:1901990","GO:0071396","GO:0000070","GO:0032993","GO:0007052","GO:2001251","GO:0051983","GO:0046827","GO:0006611","GO:0042476","GO:1904353","GO:0019904","GO:2000036","GO:0051726","GO:0045814"],["chromatin binding","chromosome organization","chromosomal region","sister chromatid segregation","regulation of chromosome organization","cellular response to heat","nuclear periphery","chromosome, centromeric region","nuclear chromosome segregation","glutamatergic synapse","SWI/SNF superfamily-type complex","mitotic spindle assembly","chromosome segregation","mitotic cell cycle process","response to heat","sister chromatid cohesion","regulation of mitotic metaphase/anaphase transition","regulation of metaphase/anaphase transition of cell cycle","metaphase/anaphase transition of mitotic cell cycle","metaphase/anaphase transition of cell cycle","mitotic sister chromatid cohesion","mitotic spindle pole","regulation of transferase activity","response to organic cyclic compound","regulation of sister chromatid segregation","mitotic cell cycle","condensed chromosome","chromatin","nuclear matrix","response to temperature stimulus","ATPase complex","RSC-type complex","positive regulation of stem cell population maintenance","regulation of cell differentiation","response to monoamine","cellular response to monoamine stimulus","response to catecholamine","cellular response to catecholamine stimulus","cellular response to organic cyclic compound","stem cell population maintenance","spindle assembly","maintenance of cell number","regulation of mitotic cell cycle phase transition","cellular response to lipid","mitotic sister chromatid segregation","protein-DNA complex","mitotic spindle organization","negative regulation of chromosome organization","regulation of chromosome segregation","positive regulation of protein export from nucleus","protein export from nucleus","odontogenesis","regulation of telomere capping","protein domain specific binding","regulation of stem cell population maintenance","regulation of cell cycle","negative regulation of gene expression, epigenetic"],[9.863238286913041e-06,1.33157537620314e-05,1.765142276782754e-05,0.0002327854961166014,0.0002327854961166014,0.0002327854961166014,0.0003256133903735207,0.0005596362026100928,0.0007920386786951619,0.0008215850797386352,0.0008215850797386352,0.001041322966586444,0.001041322966586444,0.001088980818726946,0.001088980818726946,0.001243049854266206,0.001540215927403784,0.001540215927403784,0.001700412164164208,0.001700412164164208,0.001700412164164208,0.001869627172741863,0.00198197394705453,0.002033248111958287,0.002156312784747874,0.002525082334817034,0.002519323144528129,0.003306670015605711,0.003306670015605711,0.003438503567712691,0.003637959328931511,0.003637959328931511,0.003875781927155414,0.003875781927155414,0.003875781927155414,0.003875781927155414,0.003875781927155414,0.003875781927155414,0.004100584482619278,0.004166752090441953,0.004166752090441953,0.004166752090441953,0.004166752090441953,0.004166752090441953,0.004974795074174595,0.0049300756216262,0.005270996481995121,0.005270996481995121,0.005422980944501002,0.006214345645432584,0.006344992016254061,0.007019952653474618,0.007426580512780761,0.008208629105671742,0.008360646225349584,0.008360646225349584,0.008861319676872178],[7.019057639315433e-06,1.021739376871711e-05,1.384280166736737e-05,0.0001786200856500845,0.0001786200856500845,0.0001786200856500845,0.0002553562759482004,0.0004388843358695213,0.0006077441206038274,0.0006443128954125594,0.0006443128954125594,0.0007990239967008226,0.0007990239967008226,0.000835592639392254,0.000835592639392254,0.0009538123085002639,0.001181832654791709,0.001181832654791709,0.00130475382474568,0.00130475382474568,0.00130475382474568,0.001466220512907227,0.001520800746116024,0.001560144244227084,0.001654573763079774,0.001937536618229061,0.00197573255620365,0.002593194770049352,0.002593194770049352,0.002638419540817605,0.002853002283540334,0.002853002283540334,0.00297395031622925,0.00297395031622925,0.00297395031622925,0.00297395031622925,0.00297395031622925,0.00297395031622925,0.003146444962077804,0.003197216440428962,0.003197216440428962,0.003197216440428962,0.003197216440428962,0.003197216440428962,0.003817240923788435,0.003866320575567622,0.004044521066740626,0.004044521066740626,0.00416114120916,0.00476836817569768,0.004868615254383012,0.005386523495411836,0.005698535645088862,0.005841574456223034,0.006415259411668717,0.006415259411668717,0.006799434269147354],["SMC1A/CTCF/STAG1/SMARCD1/BRD4/TPR/HDAC1/SMARCA2/HDAC2/STAG2/SUZ12/DDX5/UPF1/RAD21","SMC1A/RECQL/CTCF/NBN/STAG1/SMARCD1/ARID2/TPR/SMARCA2/STAG2/HSP90AA1/RHOA/UPF1/AURKB/XRCC1/RAD21","SMC1A/CTCF/NBN/STAG1/SMARCD1/ARID2/TPR/HDAC2/STAG2/UPF1/AURKB/XRCC1/RAD21","SMC1A/STAG1/SMARCD1/ARID2/TPR/SMARCA2/STAG2/RHOA/AURKB/RAD21","CTCF/NBN/SMARCD1/ARID2/TPR/SMARCA2/UPF1/AURKB/XRCC1/RAD21","PRKACA/TPR/HDAC2/GSK3B/HSP90AA1/DNAJB1","SMC1A/STAG1/SMARCD1/NUP93/ARID2/TPR/STAG2/RAD21","SMC1A/CTCF/STAG1/SMARCD1/ARID2/TPR/STAG2/AURKB/RAD21","SMC1A/STAG1/SMARCD1/ARID2/TPR/SMARCA2/STAG2/RHOA/AURKB/RAD21","USP8/PRKAR1A/PRKACA/EPS15/PPP2R2A/SH3GL1/GSK3B/RHOA/DNAJB1","SMARCD1/ARID2/HDAC1/SMARCA2/HDAC2/SUZ12","SMC1A/STAG1/TPR/STAG2/RHOA/AURKB","SMC1A/CTCF/STAG1/SMARCD1/ARID2/TPR/SMARCA2/STAG2/RHOA/AURKB/RAD21","SMC1A/NBN/STAG1/USP8/SMARCD1/BRD4/ARID2/TPR/SMARCA2/STAG2/RAB35/RHOA/AURKB/RAD21","PRKACA/TPR/HDAC2/GSK3B/HSP90AA1/DNAJB1","SMC1A/CTCF/STAG1/STAG2/RAD21","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","SMC1A/STAG1/STAG2/RAD21","SMC1A/STAG1/STAG2/AURKB","NBN/ABI1/PRKAR1A/MAP2K1/EIF4A2/GSK3B/HSP90AA1/RHOA/AURKB/XRCC1/CDK12","CTNNA1/GNB1/USP8/PRKACA/TPR/HDAC1/PPP2R2A/HDAC2/DDX5/GSK3B/HSP90AA1/RHOA","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","SMC1A/NBN/STAG1/USP8/SMARCD1/BRD4/ARID2/TPR/SMARCA2/STAG2/RAB35/RHOA/AURKB/RAD21","SMC1A/CTCF/SMARCD1/BRD4/ARID2/TPR/AURKB/RAD21","STAG1/SMARCD1/ARID2/HDAC1/SMARCA2/HDAC2/STAG2/SUZ12/NFKB2/UPF1/XRCC1/RAD21","SMC1A/STAG1/SMARCD1/ARID2/STAG2/RAD21","PRKACA/TPR/HDAC2/GSK3B/HSP90AA1/DNAJB1","SMARCD1/ARID2/HDAC1/SMARCA2/HDAC2/SUZ12","SMARCD1/ARID2/SUZ12","SMARCD1/HDAC1/SMARCA2/HDAC2","CTNNA1/SMARCD1/DDX6/ARID2/PRKACA/HDAC1/MAP2K1/SMARCA2/HDAC2/SUZ12/DDX5/GSK3B/RHOA/CDK12","GNB1/PRKACA/HDAC2/GSK3B","GNB1/PRKACA/HDAC2/GSK3B","GNB1/PRKACA/HDAC2/GSK3B","GNB1/PRKACA/HDAC2/GSK3B","CTNNA1/GNB1/USP8/PRKACA/HDAC1/HDAC2/DDX5/GSK3B/RHOA","SMC1A/SMARCD1/DDX6/HDAC1/SMARCA2/HDAC2","SMC1A/STAG1/TPR/STAG2/RHOA/AURKB","SMC1A/SMARCD1/DDX6/HDAC1/SMARCA2/HDAC2","NBN/SMARCD1/BRD4/ARID2/TPR/SMARCA2/AURKB/RAD21","GNB1/USP8/SMARCD1/HDAC1/HDAC2/DDX5/GSK3B/RHOA/UPF1","SMC1A/STAG1/TPR/STAG2/RHOA/AURKB/RAD21","STAG1/SMARCD1/ARID2/HDAC1/SMARCA2/HDAC2/STAG2/SUZ12/NFKB2/UPF1/XRCC1/RAD21","SMC1A/STAG1/TPR/STAG2/RHOA/AURKB","NBN/TPR/AURKB/XRCC1/RAD21","SMARCD1/ARID2/TPR/SMARCA2/AURKB/RAD21","PRKACA/TPR/GSK3B","PRKACA/TPR/GSK3B/NUP214","CTNNA1/HDAC1/HDAC2/RHOA","NBN/AURKB/XRCC1","USP8/DDX6/ABI1/PRKAR1A/PRKACA/EPS15/HDAC1/CCDC6/U2AF2/DDX5/HSP90AA1","SMARCD1/HDAC1/SMARCA2/HDAC2","CTCF/NBN/SMARCD1/BRD4/ARID2/PRKACA/TPR/SMARCA2/RHOA/NUP214/AURKB/CDK12/RAD21","CTCF/TPR/HDAC1/HDAC2/SUZ12"],[14,16,13,10,10,6,8,9,10,9,6,6,11,14,6,5,6,6,6,6,4,4,11,12,6,14,8,12,6,6,6,3,4,14,4,4,4,4,9,6,6,6,8,9,7,12,6,5,6,3,4,4,3,11,4,13,5],["Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4","Cluster 4"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Description<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Description","targets":1},{"name":"p.adjust","targets":2},{"name":"qvalue","targets":3},{"name":"geneID","targets":4},{"name":"Count","targets":5},{"name":"Cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>gene_ontology_top$cluster5 %&gt;% 
  DT::datatable(caption = &quot;Gene ontology for Cluster 5&quot;)</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-d04dbdcb8b96463c81d0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d04dbdcb8b96463c81d0">{"x":{"filter":"none","vertical":false,"caption":"<caption>Gene ontology for Cluster 5<\/caption>","data":[["GO:0006259","GO:0002520","GO:0006974","GO:0014070","GO:0032870","GO:0071407","GO:0051054","GO:0051052","GO:0071383","GO:0006281","GO:0048545","GO:0045944","GO:1901699","GO:0048511","GO:0006310","GO:0002684","GO:0009755","GO:0000122","GO:0030518","GO:0051253","GO:0009725","GO:0071417","GO:1901701","GO:0043401","GO:0006302","GO:0033993","GO:0019220","GO:0051174","GO:0042752","GO:0030522","GO:0007623","GO:0071396","GO:0030097","GO:0010628","GO:0051726","GO:0050776","GO:0050778","GO:0010562","GO:0045937","GO:0051276","GO:0071560","GO:0042327","GO:0071375","GO:0048871","GO:0071559","GO:0000003","GO:0007267","GO:0045892","GO:1902679","GO:0002521","GO:0030099","GO:0016233","GO:0033143","GO:1901698","GO:0048609","GO:0031401","GO:0009628","GO:0001775","GO:0022414","GO:0046649","GO:0051251","GO:0001934","GO:0016445","GO:0048534","GO:0046777","GO:0043434","GO:0032504","GO:0042110","GO:0010833","GO:0006468","GO:1902533","GO:0031297","GO:0048013","GO:1901653","GO:0010243","GO:0016055","GO:0060322","GO:0007417","GO:0032868","GO:0000723","GO:0000018","GO:0198738","GO:0044093","GO:0048660","GO:0048873","GO:0042325","GO:0048659","GO:0043085","GO:0002252","GO:0033002","GO:0032101","GO:0002696","GO:0032200","GO:0010629","GO:0009416","GO:0045596","GO:0033044","GO:0045321","GO:0002200","GO:0030038"],["DNA metabolic process","immune system development","DNA damage response","response to organic cyclic compound","cellular response to hormone stimulus","cellular response to organic cyclic compound","positive regulation of DNA metabolic process","regulation of DNA metabolic process","cellular response to steroid hormone stimulus","DNA repair","response to steroid hormone","positive regulation of transcription by RNA polymerase II","cellular response to nitrogen compound","rhythmic process","DNA recombination","positive regulation of immune system process","hormone-mediated signaling pathway","negative regulation of transcription by RNA polymerase II","intracellular steroid hormone receptor signaling pathway","negative regulation of RNA metabolic process","response to hormone","cellular response to organonitrogen compound","cellular response to oxygen-containing compound","steroid hormone mediated signaling pathway","double-strand break repair","response to lipid","regulation of phosphate metabolic process","regulation of phosphorus metabolic process","regulation of circadian rhythm","intracellular receptor signaling pathway","circadian rhythm","cellular response to lipid","hemopoiesis","positive regulation of gene expression","regulation of cell cycle","regulation of immune response","positive regulation of immune response","positive regulation of phosphorus metabolic process","positive regulation of phosphate metabolic process","chromosome organization","cellular response to transforming growth factor beta stimulus","positive regulation of phosphorylation","cellular response to peptide hormone stimulus","multicellular organismal-level homeostasis","response to transforming growth factor beta","reproduction","cell-cell signaling","negative regulation of DNA-templated transcription","negative regulation of RNA biosynthetic process","leukocyte differentiation","myeloid cell differentiation","telomere capping","regulation of intracellular steroid hormone receptor signaling pathway","response to nitrogen compound","multicellular organismal reproductive process","positive regulation of protein modification process","response to abiotic stimulus","cell activation","reproductive process","lymphocyte activation","positive regulation of lymphocyte activation","positive regulation of protein phosphorylation","somatic diversification of immunoglobulins","hematopoietic or lymphoid organ development","protein autophosphorylation","response to peptide hormone","multicellular organism reproduction","T cell activation","telomere maintenance via telomere lengthening","protein phosphorylation","positive regulation of intracellular signal transduction","replication fork processing","ephrin receptor signaling pathway","cellular response to peptide","response to organonitrogen compound","Wnt signaling pathway","head development","central nervous system development","response to insulin","telomere maintenance","regulation of DNA recombination","cell-cell signaling by wnt","positive regulation of molecular function","regulation of smooth muscle cell proliferation","homeostasis of number of cells within a tissue","regulation of phosphorylation","smooth muscle cell proliferation","positive regulation of catalytic activity","immune effector process","muscle cell proliferation","regulation of response to external stimulus","positive regulation of leukocyte activation","telomere organization","negative regulation of gene expression","response to light stimulus","negative regulation of cell differentiation","regulation of chromosome organization","leukocyte activation","somatic diversification of immune receptors","contractile actin filament bundle assembly"],[8.230472931917572e-08,7.825528416150813e-07,7.825528416150813e-07,1.01764024457984e-05,1.29747619730348e-05,1.29747619730348e-05,2.266297060183741e-05,2.266297060183741e-05,2.266297060183741e-05,5.27183290177535e-05,5.978648556994132e-05,6.13575503726039e-05,6.13575503726039e-05,9.266039572881553e-05,9.266039572881553e-05,9.266039572881553e-05,9.266039572881553e-05,0.0001213027644135353,0.0001240887315489254,0.000128320968603706,0.000142143054809107,0.000142143054809107,0.0001506611557807256,0.000155774889496269,0.0002097466436358676,0.0002365714231006122,0.0003280687688492706,0.0003280687688492706,0.0003300034388491848,0.0003459433705044699,0.000407092924440047,0.000407092924440047,0.0005571525869417439,0.0006458293211266267,0.0006458293211266267,0.0006458293211266267,0.0007402510836568997,0.0007402510836568997,0.0007402510836568997,0.0007479813187973009,0.0007479813187973009,0.0007755403148456871,0.0007755403148456871,0.0008180626208137386,0.0008953862452350167,0.0008953862452350167,0.0009709594204747209,0.001136102853062717,0.001204255076709847,0.001306460820618294,0.00139646405110942,0.001465165767338255,0.001612501639614347,0.001765705676701453,0.001765705676701453,0.001880319461545671,0.001905272134429413,0.001905272134429413,0.001905272134429413,0.002411363649412421,0.002412156137730456,0.002412156137730456,0.002412156137730456,0.002412156137730456,0.002453400447550346,0.002453400447550346,0.002612052985544031,0.002680382245527621,0.002683820129108812,0.002964183693355233,0.003074324704464902,0.003190707189467946,0.003190707189467946,0.003209830900004395,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003231917460234678,0.003261776811674915,0.003343863280527722,0.003528440290346869,0.003563240948213593,0.003653887201405284,0.003653887201405284,0.003653887201405284,0.003653887201405284,0.003737114550588065,0.003737114550588065,0.003737114550588065,0.003919382842180034,0.004000565724070785,0.004000565724070785,0.004019466968607481,0.004019466968607481,0.00440446371299962,0.004583424112120543],[5.974611891912996e-08,5.680657177587565e-07,5.680657177587565e-07,7.387188509396406e-06,9.41855563101566e-06,9.41855563101566e-06,1.645135762961142e-05,1.645135762961142e-05,1.645135762961142e-05,3.826894980114692e-05,4.339982047406282e-05,4.454027771517143e-05,4.454027771517143e-05,6.726343757038066e-05,6.726343757038066e-05,6.726343757038066e-05,6.726343757038066e-05,8.805532133840289e-05,9.007769265476202e-05,9.314993091446499e-05,0.000103183570693971,0.000103183570693971,0.0001093669756796297,0.0001130791043169137,0.0001522579324082324,0.0001717304035181843,0.0002381495673390714,0.0002381495673390714,0.0002395539705227074,0.000251124983028517,0.0002955142733099875,0.0002955142733099875,0.0004044446168631868,0.0004688162605074326,0.0004688162605074326,0.0004688162605074326,0.0005373582981200051,0.0005373582981200051,0.0005373582981200051,0.0005429697806167181,0.0005429697806167181,0.0005629752294994117,0.0005629752294994117,0.0005938427479287679,0.0006499730152675548,0.0006499730152675548,0.0007048326078124469,0.0008247124645804947,0.0008741850877493784,0.0009483776229813672,0.001013712188279255,0.001063583695563345,0.001170536802862007,0.001281749690558904,0.001281749690558904,0.001364949390937225,0.001383062874497578,0.001383062874497578,0.001383062874497578,0.001750441566928229,0.001751016844943029,0.001751016844943029,0.001751016844943029,0.001751016844943029,0.00178095664864107,0.00178095664864107,0.001896124677018075,0.001945725813263894,0.001948221419567092,0.002151741132086882,0.002231693985368406,0.002316177609172459,0.002316177609172459,0.002330059769934508,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002346092703460304,0.002367768011510687,0.002427355692198054,0.002561342646163079,0.002586604915543811,0.002652406259738287,0.002652406259738287,0.002652406259738287,0.002652406259738287,0.0027128221209255,0.0027128221209255,0.0027128221209255,0.002845133145027286,0.002904064899687859,0.002904064899687859,0.002917785569364441,0.002917785569364441,0.003197259925992591,0.003327169706100482],["PARP1/TOP1/MAPK1/DEK/NPM1/IGF1R/SMC3/SMARCA4/CNBP/ARID1A/FH/MYO18A/PDS5B/SRC/MRE11/ZBTB7A/KMT2A/ATRX/SAMHD1/MAPK3/TFRC/RAD50/BAX/TRIP13/UBR5/PRKDC/CHD4/DNMT1/MSH6/NONO/SFPQ/MSH2/FAF1/TOP2A/FUS/NSD2/HSP90AB1","LYN/HLA-B/MAPK1/RBM15/KMT2A/SAMHD1/CRKL/MAPK3/TFRC/PRKDC/MSH6/SBDS/MSH2/NSD2","PARP1/LYN/MAPK1/DEK/NPM1/SMC3/SMARCA4/ARID1A/FH/PDS5B/PTPN11/MRE11/YAP1/ZBTB7A/ATRX/SAMHD1/PAK1/MAPK3/RAD50/BAX/TRIP13/UBR5/PRKDC/CHD4/MSH6/NONO/UBA1/SFPQ/MSH2/TOP2A/FUS/NSD2","PARP1/CAD/LYN/MTAP/MAPK1/GNA11/NPM1/IGF1R/SMARCA4/ARID1A/GNAQ/GNAS/NCSTN/SRC/IDH1/YAP1/ZBTB7A/EIF4E/PAK1/STAT3/MAPK3/STAT1/NCOR2/ATP1A1/UBR5/DNMT1","PARP1/LYN/RAP1GDS1/MAPK1/IGF1R/SMARCA4/ARID1A/SRC/PTPN11/YAP1/ZBTB7A/EIF4E/MAX/PAK1/STAT3/MAPK3/STAT1/NCOR2/ATP1A1/UBR5/PRKDC/NONO","PARP1/MAPK1/GNA11/NPM1/IGF1R/SMARCA4/ARID1A/GNAQ/GNAS/NCSTN/SRC/YAP1/ZBTB7A/EIF4E/PAK1/STAT1/NCOR2/ATP1A1/UBR5/DNMT1","PARP1/MAPK1/IGF1R/SMARCA4/ARID1A/MRE11/ATRX/MAPK3/TFRC/RAD50/BAX/PRKDC/MSH2/FAF1/FUS/NSD2/HSP90AB1","PARP1/MAPK1/DEK/NPM1/IGF1R/SMC3/SMARCA4/ARID1A/SRC/MRE11/ATRX/MAPK3/TFRC/RAD50/BAX/PRKDC/MSH6/MSH2/FAF1/FUS/NSD2/HSP90AB1","PARP1/MAPK1/IGF1R/SMARCA4/ARID1A/SRC/YAP1/ZBTB7A/EIF4E/PAK1/NCOR2/ATP1A1/UBR5","PARP1/DEK/NPM1/SMC3/SMARCA4/ARID1A/FH/PDS5B/MRE11/ZBTB7A/ATRX/SAMHD1/RAD50/TRIP13/UBR5/PRKDC/CHD4/MSH6/NONO/SFPQ/MSH2/FUS/NSD2","PARP1/CAD/MAPK1/IGF1R/SMARCA4/ARID1A/SRC/IDH1/YAP1/ZBTB7A/EIF4E/PAK1/NCOR2/ATP1A1/UBR5","MET/PARP1/DEK/NPM1/SMARCA4/CNBP/YES1/SQSTM1/YAP1/RBM15/CDC73/KMT2A/MAX/SETD3/ATRX/STAT3/TBL1XR1/MAPK3/STAT1/PCBP1/PRKDC/SMARCA1/DDX3X/CBFB/SFPQ/TOP2A","PARP1/LYN/RAP1GDS1/MAPK1/GNA11/NPM1/IGF1R/GNAQ/GNAS/NCSTN/SRC/PTPN11/MAX/ATRX/CRKL/STAT3/MAPK3/STAT1/CUL3/PRKDC/DNMT1/NONO","PARP1/TOP1/GNA11/IGF1R/GNAQ/SRC/PPP1CB/KMT2A/NCOR2/PRKDC/NONO/SFPQ/TOP2A","PARP1/MRE11/SAMHD1/TFRC/RAD50/TRIP13/PRKDC/CHD4/MSH6/NONO/SFPQ/MSH2/TOP2A/FUS/NSD2","LYN/HLA-B/MAPK1/HLA-C/SMARCA4/ARID1A/TAP2/YES1/MYO18A/CD276/SRC/PTPN11/PUM1/CRKL/PAK1/MAPK3/TFRC/BAX/PRKDC/NONO/DDX3X/CBFB/SFPQ/MSH2/RAC1/NSD2","PARP1/MAPK1/SMARCA4/ARID1A/SRC/PTPN11/YAP1/ZBTB7A/PAK1/NCOR2/UBR5","PARP1/CUX1/SMARCA4/CNBP/RBM10/TRIM33/PHF6/SQSTM1/YAP1/ZBTB7A/CDC73/MAX/STAT3/TBL1XR1/STAT1/CUL3/NCOR2/CHD4/DNMT1/CBFB/SFPQ/NSD2","PARP1/MAPK1/SMARCA4/ARID1A/SRC/YAP1/ZBTB7A/PAK1/NCOR2/UBR5","PARP1/CUX1/NPM1/TAF15/SMARCA4/CNBP/RBM10/TRIM33/PHF6/SQSTM1/YAP1/ZBTB7A/CDC73/MAX/PTPRK/STAT3/TBL1XR1/STAT1/CUL3/NCOR2/LARP4B/CSDE1/CHD4/DNMT1/NONO/CBFB/SFPQ/FUS/NSD2","PARP1/CAD/LYN/RAP1GDS1/MAPK1/IGF1R/SMARCA4/ARID1A/SRC/IDH1/PTPN11/YAP1/ZBTB7A/EIF4E/MAX/PAK1/STAT3/MAPK3/STAT1/NCOR2/ATP1A1/UBR5/PRKDC/NONO","PARP1/LYN/RAP1GDS1/MAPK1/GNA11/IGF1R/GNAQ/GNAS/NCSTN/SRC/PTPN11/MAX/CRKL/STAT3/MAPK3/STAT1/CUL3/PRKDC/DNMT1/NONO","MET/PARP1/LYN/RAP1GDS1/MAPK1/GNA11/IGF1R/SMARCA4/YES1/SQSTM1/GNAQ/GNAS/NCSTN/SRC/PTPN11/YAP1/EIF4E/CDC73/MAX/PTPRK/CRKL/STAT3/MAPK3/STAT1/CUL3/PRKDC/DNMT1/NONO","PARP1/MAPK1/SMARCA4/ARID1A/SRC/YAP1/ZBTB7A/PAK1/NCOR2/UBR5","PARP1/DEK/SMARCA4/ARID1A/MRE11/ZBTB7A/SAMHD1/RAD50/TRIP13/PRKDC/CHD4/SFPQ/MSH2/FUS/NSD2","PARP1/CAD/LYN/MTAP/MAPK1/IGF1R/SMARCA4/ARID1A/YES1/SRC/IDH1/YAP1/ZBTB7A/EIF4E/CDC73/PAK1/STAT3/MAPK3/TFRC/NCOR2/ATP1A1/UBR5","MET/PARP1/LYN/MAPK1/NPM1/IGF1R/ACSL3/YES1/SQSTM1/GNAQ/SRC/IDH1/PTPN11/MRE11/CRKL/PAK1/MAPK3/PDK1/TFRC/RAD50/BAX/PRKDC/DDX3X/RAC1/YWHAE/HSP90AB1","MET/PARP1/LYN/MAPK1/NPM1/IGF1R/ACSL3/YES1/SQSTM1/GNAQ/SRC/IDH1/PTPN11/MRE11/CRKL/PAK1/MAPK3/PDK1/TFRC/RAD50/BAX/PRKDC/DDX3X/RAC1/YWHAE/HSP90AB1","PARP1/GNA11/GNAQ/PPP1CB/PRKDC/NONO/SFPQ/TOP2A","PARP1/MAPK1/SMARCA4/ARID1A/SRC/YAP1/ZBTB7A/PUM1/CRKL/PAK1/STAT3/NCOR2/UBR5/DDX3X","PARP1/TOP1/GNA11/GNAQ/PPP1CB/KMT2A/PRKDC/NONO/SFPQ/TOP2A","PARP1/LYN/MAPK1/IGF1R/SMARCA4/ARID1A/YES1/SRC/YAP1/ZBTB7A/EIF4E/CDC73/PAK1/MAPK3/NCOR2/ATP1A1/UBR5","PARP1/LYN/HLA-B/ATP6AP1/SMARCA4/ARID1A/SRC/PTPN11/RBM15/ZBTB7A/CDC73/KMT2A/STAT3/STAT1/TFRC/BAX/CDC42/PRKDC/CBFB/SBDS/MSH2/TOP2A","FUBP1/NPM1/TAF15/SMARCA4/CNBP/RBM10/ARID1A/CD276/SRC/PTPN11/YAP1/DAZAP1/KMT2A/STAT3/STAT1/TFRC/LARP4B/NUP98/CSDE1/UBR5/PRKDC/DNMT1/DDX3X/MSH2/FUS/NSD2","SND1/NPM1/IGF1R/SMARCA4/ARID1A/PTPN11/MRE11/CLTC/PUM1/EIF4E/CDC73/ATRX/RHEB/PTPRK/STAT3/CUL3/RAD50/BAX/TRIP13/CDC42/PRKDC/DDX3X/MSH2/PRCC/MKI67/YWHAE/HSP90AB1","PARP1/LYN/HLA-B/MAPK1/HLA-C/TAP2/YES1/CD276/SRC/PTPN11/PUM1/SAMHD1/CRKL/PAK1/TFRC/BAX/PRKDC/NONO/DDX3X/SFPQ/MSH2/NSD2","LYN/HLA-B/MAPK1/HLA-C/TAP2/YES1/CD276/SRC/PUM1/CRKL/PAK1/TFRC/BAX/PRKDC/NONO/DDX3X/SFPQ/MSH2/NSD2","MET/LYN/MAPK1/IGF1R/ACSL3/YES1/SQSTM1/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/TFRC/RAD50/DDX3X/RAC1/HSP90AB1","MET/LYN/MAPK1/IGF1R/ACSL3/YES1/SQSTM1/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/TFRC/RAD50/DDX3X/RAC1/HSP90AB1","PARP1/EML4/TOP1/MAPK1/SMC3/SMARCA4/ARID1A/PDS5B/SRC/MRE11/ATRX/MAPK3/CUL3/RAD50/NUP98/TRIP13/PRKDC/DDX3X/MSH2/TOP2A/HSP90AB1","PARP1/IGF1R/TRIM33/YES1/SRC/ZBTB7A/KMT2A/PTPRK/CRKL/STAT3/HSP90AB1","MET/LYN/MAPK1/IGF1R/YES1/SQSTM1/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/TFRC/RAD50/DDX3X/RAC1/HSP90AB1","PARP1/LYN/RAP1GDS1/MAPK1/IGF1R/SRC/PTPN11/MAX/STAT3/MAPK3/STAT1/PRKDC/NONO","LYN/IGF1R/SMARCA4/FH/SQSTM1/GNAS/NCSTN/SRC/PTPN11/YAP1/ZBTB7A/KMT2A/RHEB/STAT3/TBL1XR1/STAT1/TFRC/BAX/PRKDC","PARP1/IGF1R/TRIM33/YES1/SRC/ZBTB7A/KMT2A/PTPRK/CRKL/STAT3/HSP90AB1","PARP1/CAD/PAFAH1B2/IGF1R/SMC3/SRC/IDH1/PTPN11/MRE11/DAZAP1/PUM1/ATRX/CRKL/STAT3/NCOR2/RAD50/BAX/TRIP13/CSDE1/PRKDC/MSH6/DDX3X/MSH2/TOP2A","LYN/MAPK1/GNA11/SMARCA4/SQSTM1/GNAQ/NCSTN/HIP1/SRC/KIF5B/PTPN11/STRN/YAP1/EIF4E/CDC73/KMT2A/CRKL/TBL1XR1/MAPK3/ERC1/CUL3/UBR5/CDC42/DDX3X/RAC1/HSP90AB1","PARP1/CUX1/SMARCA4/CNBP/RBM10/TRIM33/PHF6/SQSTM1/YAP1/ZBTB7A/CDC73/MAX/PTPRK/STAT3/TBL1XR1/STAT1/CUL3/NCOR2/CHD4/DNMT1/NONO/CBFB/SFPQ/NSD2","PARP1/CUX1/SMARCA4/CNBP/RBM10/TRIM33/PHF6/SQSTM1/YAP1/ZBTB7A/CDC73/MAX/PTPRK/STAT3/TBL1XR1/STAT1/CUL3/NCOR2/CHD4/DNMT1/NONO/CBFB/SFPQ/NSD2","PARP1/LYN/HLA-B/ATP6AP1/SMARCA4/ARID1A/SRC/ZBTB7A/KMT2A/STAT3/TFRC/BAX/CDC42/PRKDC/CBFB/MSH2","PARP1/LYN/ATP6AP1/SRC/PTPN11/RBM15/ZBTB7A/CDC73/STAT3/STAT1/TFRC/CDC42/PRKDC/CBFB","MAPK1/MRE11/MAPK3/RAD50/PRKDC","PARP1/SMARCA4/SRC/YAP1/ZBTB7A/PAK1/NCOR2","PARP1/CAD/LYN/RAP1GDS1/MAPK1/GNA11/NPM1/IGF1R/GNAQ/GNAS/NCSTN/SRC/PTPN11/MAX/ATRX/CRKL/STAT3/MAPK3/STAT1/CUL3/NCOR2/PRKDC/DNMT1/NONO/FAF1","PARP1/PAFAH1B2/IGF1R/SRC/PTPN11/DAZAP1/PUM1/ATRX/CRKL/NCOR2/BAX/TRIP13/MSH6/DDX3X/MSH2/TOP2A","LYN/MAPK1/DEK/NPM1/YES1/SQSTM1/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/ASPSCR1/TFRC/CUL3/RAD50/DDX3X/RAC1/HSP90AB1","PARP1/LYN/ATP6AP1/GNA11/NPM1/IGF1R/GNAQ/SRC/YAP1/PPP1CB/KMT2A/PTPRK/MAPK3/STAT1/PDK1/TFRC/BAX/PRKDC/MSH6/NONO/DDX3X/MSH2/YWHAE/HSP90AB1","LYN/SMARCA4/ARID1A/YES1/SDC4/GNAQ/MYO18A/CD276/GNAS/NCSTN/SRC/PTPN11/ZBTB7A/KMT2A/STAT3/GNA13/TFRC/BAX/PRKDC/MSH6/CBFB/MSH2/NSD2","PARP1/CAD/PAFAH1B2/IGF1R/SMC3/SRC/IDH1/PTPN11/MRE11/DAZAP1/PUM1/ATRX/CRKL/NCOR2/RAD50/BAX/TRIP13/CSDE1/PRKDC/MSH6/DDX3X/MSH2/TOP2A","LYN/SMARCA4/ARID1A/YES1/SDC4/CD276/NCSTN/SRC/PTPN11/ZBTB7A/KMT2A/STAT3/TFRC/BAX/PRKDC/MSH6/CBFB/MSH2/NSD2","LYN/SMARCA4/ARID1A/YES1/CD276/SRC/PTPN11/TFRC/PRKDC/CBFB/MSH2/NSD2","LYN/MAPK1/YES1/SQSTM1/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/TFRC/RAD50/DDX3X/RAC1/HSP90AB1","SAMHD1/TFRC/PRKDC/MSH6/MSH2/NSD2","MAPK1/RBM15/KMT2A/CRKL/MAPK3/SBDS","CAD/LYN/IGF1R/SRC/MRE11/PAK1/MAPK3/RAD50/DDX3X","PARP1/CAD/LYN/RAP1GDS1/MAPK1/IGF1R/SRC/PTPN11/MAX/STAT3/MAPK3/STAT1/PRKDC/NONO","PARP1/PAFAH1B2/IGF1R/SRC/PTPN11/DAZAP1/PUM1/ATRX/CRKL/NCOR2/BAX/TRIP13/MSH6/DDX3X/MSH2/TOP2A","LYN/SMARCA4/ARID1A/YES1/SDC4/CD276/NCSTN/SRC/PTPN11/KMT2A/STAT3/TFRC/BAX/PRKDC/CBFB","PARP1/MAPK1/SRC/MRE11/MAPK3/RAD50/HSP90AB1","CAD/LYN/TOP1/MAPK1/NPM1/IGF1R/YES1/SQSTM1/GNAQ/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/ERC1/PDK1/TFRC/RAD50/BAX/PRKDC/DDX3X/RAC1/HSP90AB1","PARP1/LYN/IGF1R/HIP1/SRC/CANT1/PTPN11/RHEB/CRKL/PAK1/MAPK3/TFRC/CUL3/BAX/TFG/CDC42/DDX3X/SFPQ/RAC1/HSP90AB1","PARP1/MRE11/ATRX/SAMHD1/RAD50","LYN/YES1/SRC/PTPN11/PAK1","PARP1/LYN/RAP1GDS1/MAPK1/IGF1R/SRC/PTPN11/MAX/STAT3/MAPK3/STAT1/PRKDC/NONO","PARP1/CAD/LYN/RAP1GDS1/MAPK1/GNA11/IGF1R/GNAQ/GNAS/NCSTN/SRC/PTPN11/MAX/CRKL/STAT3/MAPK3/STAT1/CUL3/NCOR2/PRKDC/DNMT1/NONO/FAF1","SMARCA4/GNAQ/SRC/STRN/YAP1/CDC73/TBL1XR1/CUL3/UBR5/CDC42/DDX3X/RAC1","ATIC/MAPK1/IGF1R/PCM1/NCSTN/SRC/PTPN11/DDX10/ATRX/CRKL/MAPK3/NCOR2/BAX/CDC42/PRKDC/SMARCA1/YWHAE","LYN/ATIC/MAPK1/IGF1R/PCM1/NCSTN/SRC/PTPN11/ATRX/RHEB/CRKL/STAT3/MAPK3/NCOR2/BAX/CDC42/PRKDC/SMARCA1/YWHAE/HSP90AB1","PARP1/CAD/LYN/MAPK1/IGF1R/SRC/PTPN11/MAX/MAPK3/STAT1/PRKDC","PARP1/MAPK1/SRC/MRE11/ATRX/MAPK3/RAD50/PRKDC/HSP90AB1","PARP1/MRE11/TFRC/RAD50/MSH6/MSH2/FUS/NSD2","SMARCA4/GNAQ/SRC/STRN/YAP1/CDC73/TBL1XR1/CUL3/UBR5/CDC42/DDX3X/RAC1","MET/LYN/RAP1GDS1/MAPK1/NPM1/IGF1R/SMARCA4/GNAQ/NCSTN/HIP1/SRC/MRE11/ZBTB7A/PAK1/STAT3/MAPK3/ERC1/GNA13/TFRC/RAD50/BAX/MSH6/DDX3X/MSH2/HSP90AB1","IGF1R/RBM10/SRC/STAT1/GNA13/PRKDC/DNMT1","FH/PTPN11/KMT2A/BAX","MET/LYN/MAPK1/NPM1/IGF1R/YES1/SQSTM1/GNAQ/SRC/PTPN11/MRE11/CRKL/PAK1/MAPK3/TFRC/RAD50/BAX/PRKDC/DDX3X/RAC1/HSP90AB1","IGF1R/RBM10/SRC/STAT1/GNA13/PRKDC/DNMT1","MET/LYN/RAP1GDS1/MAPK1/IGF1R/GNAQ/NCSTN/HIP1/SRC/MRE11/PAK1/STAT3/MAPK3/GNA13/RAD50/BAX/MSH6/DDX3X/MSH2/HSP90AB1","LYN/HLA-B/HLA-C/TAP2/YES1/MYO18A/SRC/KIF5B/KMT2A/PAK1/STAT3/TFRC/MSH6/MSH2/NSD2","IGF1R/RBM10/SRC/YAP1/STAT1/GNA13/PRKDC/DNMT1","MET/PARP1/LYN/HLA-B/MAPK1/IGF1R/YES1/SRC/PTPN11/PUM1/SAMHD1/PAK1/STAT3/MAPK3/STAT1/BAX/PRKDC/NONO/DDX3X/SFPQ/RAC1","LYN/SMARCA4/ARID1A/YES1/CD276/SRC/PTPN11/TFRC/PRKDC/CBFB/MSH2/NSD2","PARP1/MAPK1/SRC/MRE11/ATRX/MAPK3/RAD50/PRKDC/HSP90AB1","SND1/NPM1/TAF15/RBM10/SRC/PTPN11/YAP1/PUM1/EIF4E/MAX/ATRX/CRKL/DIS3/STAT3/CUL3/LARP4B/CSDE1/UBR5/CHD4/DNMT1/DDX3X/FUS/RAC1/HSP90AB1","PARP1/GNA11/NPM1/GNAQ/PPP1CB/KMT2A/PTPRK/BAX/MSH6/MSH2","LYN/MAPK1/SMARCA4/PCM1/PTPN11/YAP1/RBM15/EIF4E/CDC73/STAT3/STAT1/DNMT1/CBFB","PARP1/MAPK1/SMARCA4/ARID1A/SRC/MRE11/ATRX/MAPK3/CUL3/RAD50/TRIP13/TOP2A","LYN/SMARCA4/ARID1A/YES1/SDC4/MYO18A/CD276/NCSTN/SRC/PTPN11/ZBTB7A/KMT2A/STAT3/TFRC/BAX/PRKDC/MSH6/CBFB/MSH2/NSD2","SAMHD1/TFRC/PRKDC/MSH6/MSH2/NSD2","MET/SDC4/SRC/PAK1/CUL3/CDC42/RAC1"],[37,14,32,26,22,20,17,22,13,23,15,26,22,13,15,26,11,22,10,29,24,20,28,10,15,22,26,26,8,14,10,17,22,26,27,22,19,18,18,21,11,17,13,19,11,24,26,24,24,16,14,5,7,25,16,19,24,23,23,19,12,15,6,6,9,14,16,15,7,24,20,5,5,13,23,12,17,20,11,9,8,12,25,7,4,21,7,20,15,8,21,12,9,24,10,13,12,20,6,7],["Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5","Cluster 5"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Description<\/th>\n      <th>p.adjust<\/th>\n      <th>qvalue<\/th>\n      <th>geneID<\/th>\n      <th>Count<\/th>\n      <th>Cluster<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Description","targets":1},{"name":"p.adjust","targets":2},{"name":"qvalue","targets":3},{"name":"geneID","targets":4},{"name":"Count","targets":5},{"name":"Cluster","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>cluster_full = clusters %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Gene&quot;) %&gt;%
  inner_join(matrix_drivers %&gt;% melt() %&gt;%
               dplyr::rename(&quot;Gene&quot; = &quot;Var1&quot;,  &quot;Correlation&quot; = &quot;value&quot; ), by = &quot;Gene&quot;) %&gt;%
  dplyr::select(-Cluster)

write.csv(cluster_full, &quot;output/Publication/Data/clusters.csv&quot;)</code></pre>
</div>
</div>
</div>
<div id="figure-3" class="section level1">
<h1>Figure 3</h1>
<p>Integration of protein reproducibility ranks and gene-specific
mRNA-protein correlations. Protein reproducibility ranks were retrieved
from <a
href="https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(22)00170-9?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS2667237522001709%3Fshowall%3Dtrue">Upadhya
et al (2022)</a>. ## Figure 3A</p>
<pre class="r"><code>correlation_overall= read.csv(&quot;output/correlations-datasets/correlation-all-nontransformed.csv&quot;,
                            header=TRUE, row.names = 1) %&gt;%
  drop_na()

ranks = read_excel(&quot;data/Ranks/1-s2.0-S2667237522001709-mmc3.xlsx&quot;, sheet=3) %&gt;%
  as.data.frame() %&gt;%
  dplyr::rename(&quot;Gene&quot; = &quot;...1&quot;) %&gt;%
  mutate(Bin = case_when(
    between(`Aggregated Reproducibility Rank`, 0, 0.2) ~ &quot;0 - 0.2&quot;,
    between(`Aggregated Reproducibility Rank`, 0.2, 0.4) ~ &quot;0.2 - 0.4&quot;,
    between(`Aggregated Reproducibility Rank`, 0.4, 0.6) ~ &quot;0.4 - 0.6&quot;,
    between(`Aggregated Reproducibility Rank`, 0.6, 0.8) ~ &quot;0.6 - 0.8&quot;,
    between(`Aggregated Reproducibility Rank`, 0.8, 1) ~ &quot;0.8 - 1&quot;
  )) %&gt;%
  mutate(Bin = factor(Bin, levels = c(&quot;-1 - -0.8&quot;,&quot;-0.8 - -0.6&quot;,
                                  &quot;-0.6 - -0.4&quot;, &quot;-0.4 - -0.2&quot;,
                                  &quot;-0.2 - 0&quot;,&quot;0 - 0.2&quot;,
                                  &quot;0.2 - 0.4&quot;, &quot;0.4 - 0.6&quot;,
                                  &quot;0.6 - 0.8&quot;,
                                  &quot;0.8 - 1&quot;)))


ranks_plot = correlation_overall %&gt;%
  dplyr::rename(&quot;Gene&quot; = &quot;Protein&quot;) %&gt;%
  inner_join(ranks, by = &quot;Gene&quot;) %&gt;%
  drop_na(&quot;Aggregated Reproducibility Rank&quot;) %&gt;%
    ggplot(aes(x=Bin, y = `Correlation`)) + geom_boxplot(fill=&quot;#be95c4&quot;) + theme_bw() +
  labs(x=&quot;Aggregated protein reproducibility rank\n(binned)&quot;,
       y=&quot;mRNA-protein correlation&quot;) +
  #annotate(&quot;text&quot;, x = 1, y = 0.9, size=6,
  #label = &quot;paste(italic(r), \&quot; = 0.48\&quot;)&quot;, parse = TRUE, color = &quot;black&quot;) +
  theme(axis.text.x = element_text( color = &quot;black&quot;, size =12),
        axis.text.y = element_text(color = &quot;black&quot;, size =12),
        axis.title = element_text(color = &quot;black&quot;, size =15)) +
    theme(
        axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
         axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=20, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
              legend.position = &quot;top&quot;) +  ggtitle(&quot;ProCan-DepMapSanger&quot;)


correlation_overall %&gt;%
  dplyr::rename(&quot;Gene&quot; = &quot;Protein&quot;) %&gt;%
  inner_join(ranks, by = &quot;Gene&quot;) %&gt;%
  drop_na(&quot;Aggregated Reproducibility Rank&quot;) %&gt;%
  with(cor.test(.$Correlation, .$`Aggregated Reproducibility Rank`, method = &quot;spearman&quot;))</code></pre>
<pre><code>
    Spearman&#39;s rank correlation rho

data:  .$Correlation and .$`Aggregated Reproducibility Rank`
S = 7150068182, p-value &lt; 2.2e-16
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.4904966 </code></pre>
<pre class="r"><code>ggsave(filename = &quot;output/Publication/Figures/Fig_3A_upper.png&quot;, height =3.5, width =7, plot = ranks_plot, dpi = 1200)</code></pre>
<pre class="r"><code>library(ggpubr)

correlation_ranks = correlation_overall %&gt;%
  dplyr::rename(&quot;Gene&quot; = &quot;Protein&quot;) %&gt;%
  inner_join(ranks, by = &quot;Gene&quot;) %&gt;%
  drop_na(&quot;Aggregated Reproducibility Rank&quot;) %&gt;%
 ggplot(aes(x = `Aggregated Reproducibility Rank`, y = Correlation)) +
  geom_point(size =3) + geom_smooth(method = &quot;lm&quot;, color = &quot;blue&quot;, fill = &quot;lightgray&quot;, se = TRUE) +
    theme_bw() + ylab(&quot;mRNA-protein correlation&quot;) +
  xlab(&quot;Aggregated protein reproducibility rank&quot;) +
  annotate(&quot;text&quot;, x = 0.2, y = 0.85, size = 6,
  label = &quot;paste(italic(r), \&quot; = 0.49, \&quot;, italic(p), \&quot; &lt; 2.2e-16\&quot;)&quot;,
  parse = TRUE, color = &quot;blue&quot;) +
  theme(axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=20, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;,
        strip.text.y = element_text(
        size = 16, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 16, family = &quot;serif&quot;
        ))




ggsave(filename = &quot;output/Publication/Figures/Fig_3A_lower.png&quot;, height =3.5, width =7, plot = correlation_ranks, dpi = 1200)</code></pre>
<pre class="r"><code>ggarrange(ranks_plot, correlation_ranks, nrow = 2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-33-1.png" alt="Association between aggregated protein reproducibility ranks from Upadhya et al 22 and mRNA-protein correlations in the ProCan-DepMapSanger dataset." width="384" />
<p class="caption">
Association between aggregated protein reproducibility ranks from
Upadhya et al 22 and mRNA-protein correlations in the
ProCan-DepMapSanger dataset.
</p>
</div>
<div id="figure-3b" class="section level2">
<h2>Figure 3B</h2>
<pre class="r"><code>drivers = read.delim(here::here(&quot;data/OncoKb/cancerGeneList.tsv&quot;), fill=TRUE, check.names = FALSE)


ranks_all_1 = correlation_overall %&gt;%
dplyr::rename(&quot;Gene&quot;= &quot;Protein&quot;) %&gt;%
  inner_join(ranks, by = &quot;Gene&quot;)</code></pre>
<pre class="r"><code>library(ggpubr)
drivers_of_interest_1 = c(&quot;CTNNB1&quot;, &quot;EGFR&quot;,&quot;IDH2&quot;,&quot;RAC2&quot;,&quot;LCP1&quot;,&quot;ALDH2&quot;, &quot;CPS1&quot;, &quot;KMT2D&quot;, &quot;DPYD&quot;)
drivers_of_interest = c(&quot;CTNNB1&quot;, &quot;EGFR&quot;,&quot;IDH2&quot;,&quot;ALDH2&quot;,&quot;KMT2D&quot;)

ranks_all_1 %&lt;&gt;%
  mutate(Drivers = ifelse(Gene %in% drivers$`Hugo Symbol`, &quot;Cancer Driver&quot;, &quot;Non-Driver&quot;)) %&gt;%
  dplyr::select(-c(`Ovarian Reproducibility Rank`, `Colon Reproducibility Rank`, `CCLE Reproducibility Rank`)) %&gt;%
  dplyr::select(Gene, everything())


ranks_v_correlation = ranks_all_1 %&gt;%
  mutate(Drivers = ifelse(Gene %in% drivers$`Hugo Symbol`, &quot;Cancer Driver&quot;, &quot;Non-Driver&quot;),
         Gene_label = ifelse(Gene %in% drivers_of_interest, Gene, NA)) %&gt;%

    ggscatter(.,x=&quot;Correlation&quot;, y =&quot;Aggregated Reproducibility Rank&quot;, add = &quot;reg.line&quot;,
               color = &quot;Drivers&quot;,size = &quot;Drivers&quot;, shape = &quot;Drivers&quot;, add.params = list(color = &quot;blue&quot;,
                                                 fill = &quot;lightgray&quot;), alpha = &quot;Drivers&quot;,  cor.method = &quot;spearman&quot;, cor.coef.size = 5.5) +
    theme_bw() + xlab(&quot;mRNA-protein correlation&quot;) +
  ylab(&quot;Aggregated protein reproducibility rank&quot;) +
  annotate(&quot;text&quot;, x = -0.2, y = 0.95, size = 6,
  label = &quot;paste(italic(r), \&quot; = 0.49, \&quot;, italic(p), \&quot; &lt; 2.2e-16\&quot;)&quot;,
  parse = TRUE, color = &quot;blue&quot;)+


  geom_vline(xintercept = 0.5, size = 0.7, lty = &quot;dashed&quot;, color = &quot;grey60&quot;) + theme(legend.position = &quot;none&quot;) +
  geom_hline(yintercept = 0.4 , size = 0.7, lty = &quot;dashed&quot;, color = &quot;grey60&quot;) +
  theme(axis.text.x = element_text(size = 26, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 26, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=26, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;, size =22),
          legend.title = element_text( family = &quot;serif&quot;, size =22),
           plot.title = element_text(family = &quot;serif&quot;, size =23),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;,
        strip.text.y = element_text(
        size = 16, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 16, family = &quot;serif&quot;
        )) + scale_shape_manual(values = c(&quot;Cancer Driver&quot; = 17,
                                           &quot;CTNNB1&quot; = 17,
                                           &quot;CPS1&quot; = 17,
                                           &quot;KMT2D&quot; = 17,
                                           &quot;MYH11&quot; = 17,
                                            &quot;DPYD&quot; = 17,
                                           &quot;Non-Driver&quot; = 19)) +
  scale_size_manual(values = c(&quot;Cancer Driver&quot; = 6,
                                           &quot;CTNNB1&quot; = 6,
                                           &quot;MYH11&quot; = 6,
                                           &quot;CPS1&quot; = 6,
                                           &quot;KMT2D&quot; = 6,
                                            &quot;DPYD&quot; = 6,
                                           &quot;Non-Driver&quot; = 3)) +
  scale_color_manual(values = c(&quot;Non-Driver&quot; = &quot;lightgrey&quot;,
                                &quot;Cancer Driver&quot; = &quot;#ad2831&quot;,
                                &quot;CTNNB1&quot; = &quot;#ad2831&quot;,
                                           &quot;CPS1&quot; = &quot;#ad2831&quot;,
                                           &quot;KMT2D&quot; = &quot;#ad2831&quot;,
                                &quot;DPYD&quot; = &quot;#ad2831&quot;,
                                &quot;MYH11&quot; = &quot;#ad2831&quot;)) +
  theme(legend.position = &quot;top&quot;) +
    scale_alpha_manual(values = c(&quot;Non-Driver&quot; = 0.45,
                                &quot;Cancer Driver&quot; = 1,
                                &quot;CTNNB1&quot; = 1,
                                           &quot;TPR&quot; = 1,
                                           &quot;EGFR&quot; = 1,
                                           &quot;LPP&quot; = 1,
                                           &quot;MTOR&quot; = 1,
                                           &quot;IDH2&quot; = 1,
                                &quot;MYH11&quot; = 1)) +
  geom_text(
    data = . %&gt;% dplyr::filter(!is.na(Gene_label)),  # Filter data to only include rows with labels
    aes(label = Gene_label),  # Use the Gene_label column for labels
    vjust = -0.3,
    hjust = -0.001,
    color = &quot;black&quot;,
    size = 6.5,
    family = &quot;serif&quot;,
  ) + labs(alpha = &quot;Gene&quot;,
           shape = &quot;Gene&quot;,
           colour = &quot;Gene&quot;,
           size = &quot;Gene&quot;)

ranks_v_correlation</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-35-1.png" alt=" Scatter plot of mRNA-protein correlation against aggregated protein reproducibility rank, with cancer drivers indicated by triangle data points." width="576" />
<p class="caption">
Scatter plot of mRNA-protein correlation against aggregated protein
reproducibility rank, with cancer drivers indicated by triangle data
points.
</p>
</div>
<pre class="r"><code>ggsave(filename = &quot;output/Publication/Figures/Fig_3B.png&quot;, height =9, width = 10, plot = ranks_v_correlation, dpi = 1200)</code></pre>
</div>
<div id="figure-3c" class="section level2">
<h2>Figure 3C</h2>
<pre class="r"><code>correlation_data_tissue = read.csv(&quot;output/correlations-datasets/correlation-tissue-nontransformed.csv&quot;, row.names = 1)
ranks_all = correlation_data_tissue %&gt;%
  inner_join(ranks, by = &quot;Gene&quot;)</code></pre>
<pre class="r"><code>cr_correlations &lt;- read.csv(&quot;output/correlations-datasets/correlation-tissue-cptac-linkedomics.csv&quot;, row.names = 1) %&gt;%
  mutate(Protein =  gsub(&quot;\\.\\d+$&quot;, &quot;&quot;, Protein)) %&gt;%
   mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein,
                        keytype = &quot;ENSEMBL&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;))

ccle = read.csv(&quot;output/correlations-datasets/CCLE-correlations-tissue.csv&quot;, row.names = 1)</code></pre>
<ul>
<li>ALDH2</li>
</ul>
<pre class="r"><code>ALDH2  = ranks_all %&gt;%
  ggplot(aes(Correlation)) +
   geom_density(data=ranks_all, color=&quot;darkgrey&quot;, size = 1) +
      geom_density(data=dplyr::filter(cr_correlations, Gene == &quot;ALDH2&quot;), color=&quot;#e3a3b3&quot;, size = 1.5, lty =&quot;dashed&quot;, alpha = 0.5) +
   geom_density(data=dplyr::filter(ccle, Protein == &quot;ALDH2&quot;), color=&quot;#6e0b24&quot;, size = 1.5, lty =&quot;dotted&quot;, alpha =0.5) +
  geom_density(data=dplyr::filter(ranks_all, Gene == &quot;ALDH2&quot;), color=&quot;#fb6f92&quot;, size = 2) +


  theme_classic() + labs(x= &quot;mRNA-protein correlation&quot;,
                    y= &quot;Density&quot;,
                    title = &quot;ALDH2&quot;) +
theme(
  axis.text = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  axis.title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;bold&quot;)
)

ggsave(plot = ALDH2, filename = &quot;output/Publication/Figures/ALDH2-dist.png&quot;, height = 5, width =7, dpi=1200)</code></pre>
<ul>
<li>IDH2</li>
</ul>
<pre class="r"><code>idh2 = ranks_all %&gt;%
  ggplot(aes(Correlation)) +
   geom_density(data=ranks_all, color=&quot;lightgray&quot;, size = 1.5) +
        geom_density(data=dplyr::filter(ccle, Protein == &quot;IDH2&quot;), color=&quot;#6e0b24&quot;, size = 1.5, lty =&quot;dotted&quot;) +
          geom_density(data=dplyr::filter(cr_correlations, Gene == &quot;IDH2&quot;), color=&quot;#e3a3b3&quot;, size = 1.5, lty =&quot;dashed&quot;) +
  geom_density(data=dplyr::filter(ranks_all, Gene == &quot;IDH2&quot;), color=&quot;#fb6f92&quot;, size = 2) +


  theme_classic() + labs(x= &quot;mRNA-protein correlation&quot;,
                    y= &quot;Density&quot;,
                    title = &quot;IDH2&quot;) +
  theme(legend.position = &quot;top&quot;,
                          legend.title = element_blank(),
        legend.spacing.y = unit(0, &quot;mm&quot;)) +
theme(
  axis.text = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  axis.title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;bold&quot;)
)

ggsave(filename = &quot;output/Publication/Figures/IDH2_distribution.png&quot;, height =5, width =7, plot = idh2, dpi = 1200)</code></pre>
<ul>
<li>EGFR</li>
</ul>
<pre class="r"><code>egfr = ranks_all %&gt;%
  ggplot(aes(Correlation)) +
   geom_density(data=ranks_all, color=&quot;lightgray&quot;, size = 1) +
        geom_density(data=dplyr::filter(cr_correlations, Gene == &quot;EGFR&quot;), color=&quot;#e3a3b3&quot;, size = 1.5, lty =&quot;dashed&quot;) +
   geom_density(data=dplyr::filter(ccle, Protein == &quot;EGFR&quot;), color=&quot;#6e0b24&quot;, size = 1.5, lty =&quot;dotted&quot;) +
  geom_density(data=dplyr::filter(ranks_all, Gene == &quot;EGFR&quot;), color=&quot;#fb6f92&quot;, size = 2) +


  theme_classic() + labs(x= &quot;mRNA-protein correlation&quot;,
                    y= &quot;Density&quot;,
                    title = &quot;EGFR&quot;) +
    theme(legend.position = &quot;top&quot;,
                          legend.title = element_blank(),
        legend.spacing.y = unit(0, &quot;mm&quot;)) +
theme(
  axis.text = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  axis.title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;bold&quot;)
)


ggsave(filename = &quot;output/Publication/Figures/EGFR_distribution.png&quot;, height =5, width =7, plot = egfr, dpi = 1200)</code></pre>
<ul>
<li>CTNNB1</li>
</ul>
<pre class="r"><code>ctnnb1 = ranks_all %&gt;%
  ggplot(aes(Correlation)) +
   geom_density(data=ranks_all, color=&quot;darkgrey&quot;, size = 1) +
      geom_density(data=dplyr::filter(cr_correlations, Gene == &quot;CTNNB1&quot;), color=&quot;#e3a3b3&quot;, size = 1.5, lty =&quot;dashed&quot;, alpha = 0.5) +
   geom_density(data=dplyr::filter(ccle, Protein == &quot;CTNNB1&quot;), color=&quot;#6e0b24&quot;, size = 1.5, lty =&quot;dotted&quot;, alpha =0.5) +
  geom_density(data=dplyr::filter(ranks_all, Gene == &quot;CTNNB1&quot;), color=&quot;#fb6f92&quot;, size = 2) +


  theme_classic() + labs(x= &quot;mRNA-protein correlation&quot;,
                    y= &quot;Density&quot;,
                    title = &quot;CTNNB1&quot;) +
theme(
  axis.text = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  axis.title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;bold&quot;)
)


ggsave(plot = ctnnb1, filename = &quot;output/Publication/Figures/CTNNB1-dist.png&quot;, height = 5, width =7, dpi=1200)</code></pre>
<ul>
<li>KMT2D</li>
</ul>
<pre class="r"><code>kmt2d  = ranks_all %&gt;%
  ggplot(aes(Correlation)) +
   geom_density(data=ranks_all, color=&quot;darkgrey&quot;, size = 1) +
      geom_density(data=dplyr::filter(cr_correlations, Gene == &quot;KMT2D&quot;), color=&quot;#e3a3b3&quot;, size = 1.5, lty =&quot;dashed&quot;, alpha = 0.5) +
   geom_density(data=dplyr::filter(ccle, Protein == &quot;KMT2D&quot;), color=&quot;#6e0b24&quot;, size = 1.5, lty =&quot;dotted&quot;, alpha =0.5) +
  geom_density(data=dplyr::filter(ranks_all, Gene == &quot;KMT2D&quot;), color=&quot;#fb6f92&quot;, size = 2) +


  theme_classic() + labs(x= &quot;mRNA-protein correlation&quot;,
                    y= &quot;Density&quot;,
                    title = &quot;KMT2D&quot;) +
theme(
  axis.text = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  axis.title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;plain&quot;),
  title = element_text(color = &quot;black&quot;, size = 25, family = &quot;serif&quot;, face = &quot;bold&quot;)
)



ggsave(plot = kmt2d, filename = &quot;output/Publication/Figures/KMT2D-dist.png&quot;, height = 5, width =7, dpi=1200)</code></pre>
<pre class="r"><code>ggarrange(ALDH2,idh2 ,
          egfr, ctnnb1,
          kmt2d, ncol = 2, nrow = 3)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-43-1.png" alt="For each gene, plots show the distribution of mRNA-protein correlations across tissues in the ProCan-DepMapSanger (pink solid line), Clinical Proteomic Tumor Analysis Consortium (CPTAC; light dashed line) and Cancer Cell Line Encyclopedia (CCLE; dark dashed line) datasets. Background (grey solid line) indicates the median mRNA-protein correlation from all genes in the ProCan-DepMapSanger dataset." width="960" />
<p class="caption">
For each gene, plots show the distribution of mRNA-protein correlations
across tissues in the ProCan-DepMapSanger (pink solid line), Clinical
Proteomic Tumor Analysis Consortium (CPTAC; light dashed line) and
Cancer Cell Line Encyclopedia (CCLE; dark dashed line) datasets.
Background (grey solid line) indicates the median mRNA-protein
correlation from all genes in the ProCan-DepMapSanger dataset.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-43-1">
Past versions of unnamed-chunk-43-1.png
</button>
</p>
<div id="fig-unnamed-chunk-43-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/urwahnawaz/OnCorr/blob/0462df41c2a09e5ded2a36493174bfe6c2de2586/docs/figure/publication-figures.Rmd/unnamed-chunk-43-1.png" target="_blank">0462df4</a>
</td>
<td>
Urwah Nawaz
</td>
<td>
2025-06-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="figure-3d" class="section level2">
<h2>Figure 3D</h2>
<pre class="r"><code>correlation_data_tissue = read.csv(&quot;output/correlations-datasets/correlation-tissue-nontransformed.csv&quot;, row.names = 1)


ctnn1_tissues = correlation_data_tissue %&gt;%
  dplyr::filter(Gene == &quot;CTNNB1&quot;) %&gt;%
  ggplot(aes(x= reorder(Tissue, Correlation), y = Correlation)) + geom_bar(stat= &quot;identity&quot;, fill= &quot;red&quot;) + geom_hline(yintercept =  0.103 , size = 0.7, lty = &quot;dashed&quot;, color = &quot;blue&quot;) +

  coord_flip() +
  theme_bw() +    ggtitle(&quot;CTNNB1&quot;) + theme(axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 24, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=24, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;none&quot;,
        strip.text.y = element_text(
        size = 10, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 10, family = &quot;serif&quot;
        )) + labs(x= &quot;&quot;, y=&quot;Spearman Correlation&quot;) +
  annotate(&quot;text&quot;, x = 4, y = 0.16,
  label = &quot;paste(italic(median), \&quot; = 0.10\&quot;)&quot;, parse = TRUE, color = &quot;blue&quot;, size = 7, angle = 270)

ctnn1_tissues</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-44-1.png" alt="mRNA-protein correlation of CTNNB1 for each tissue type. " width="480" />
<p class="caption">
mRNA-protein correlation of CTNNB1 for each tissue type.
</p>
</div>
<pre class="r"><code>#ggsave(filename = &quot;output/Publication/Figures/Fig_3F.png&quot;,height = 10, width = , plot =ctnn1_tissues, dpi = 1200)</code></pre>
</div>
<div id="figure-3e" class="section level2">
<h2>Figure 3E</h2>
<pre class="r"><code>file_dir = file.path(&quot;/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/&quot;)

data = list()

## Metadata
data$sample_md = read.delim(&quot;~/Documents/2023/RNA-protein-correlationx/Data/e0022_diann_051021_sample_mapping_averaged.txt&quot;)
data$sample_md_rep = read.table(file.path(file_dir, &quot;e0022_diann_051021_sample_mapping_replicates.txt&quot;), header=TRUE, sep = &quot;\t&quot;)


data$prt_mtx =  read.delim(file.path(file_dir, &quot;e0022_diann_051021_working_matrix_averaged.txt&quot;),
                           header = TRUE, fill = TRUE, check.names = FALSE)

data$rna_mtx = read.delim(gzfile(file.path(file_dir, &quot;rnaseq_voom.csv.gz&quot;)), header = TRUE,
                          fill = TRUE, sep = &quot;,&quot;,
                          row.names = 1, check.names = FALSE)</code></pre>
<pre class="r"><code>library(ggbeeswarm)

rna_swarm_ctnnb1 =data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;CTNNB1&quot;) %&gt;%
  scale() %&gt;%
  as.data.frame() %&gt;%
  mutate(Outlier = ifelse(CTNNB1 &gt; 3, &quot;Yes&quot;, &quot;No&quot;)) %&gt;%
  rownames_to_column(&quot;SIDM&quot;) %&gt;%
  left_join(data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;RNA&quot; = &quot;CTNNB1&quot;) %&gt;%
  rownames_to_column(&quot;SIDM&quot;), by = &quot;SIDM&quot;) %&gt;%
  ggplot(aes(y = RNA, x=&#39;&#39;, color = Outlier)) +
  geom_beeswarm(cex=3, size=3) +
   scale_color_manual(values = c(&quot;No&quot; = &quot;#8d99ae&quot;,
                                &quot;Yes&quot; = &quot;#bc4749&quot;))  +
  theme(legend.position = &quot;top&quot;) + theme_bw() +
    theme(axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=20, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;, size=20),
          legend.title = element_text( family = &quot;serif&quot;, size =20),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;top&quot;,
        strip.text.y = element_text(
        size = 10, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 10, family = &quot;serif&quot;
        )) + ggtitle(&quot;CTNNB1&quot;) + xlab(&quot;Samples&quot;) + ylab(&quot;RNA expression&quot;) + labs(color = &quot;Outlier in RNA-seq&quot;)


rna_swarm_ctnnb1</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-46-1.png" alt="Distribution of CTNNB1 mRNA expression across samples from the ProCan-DepMapSanger dataset, with outliers (&gt; 3 standard deviations above the mean) indicated" width="576" />
<p class="caption">
Distribution of CTNNB1 mRNA expression across samples from the
ProCan-DepMapSanger dataset, with outliers (&gt; 3 standard deviations
above the mean) indicated
</p>
</div>
<pre class="r"><code>ggsave(filename = &quot;output/Publication/Figures/Fig_3E_RNA.png&quot;,height = 6, width = 5, plot = rna_swarm_ctnnb1, dpi = 1200 )</code></pre>
</div>
<div id="figure-3e-1" class="section level2">
<h2>Figure 3E</h2>
<pre class="r"><code>ctnnb1= data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;CTNNB1&quot;) %&gt;%
  scale() %&gt;%
  as.data.frame() %&gt;%
  mutate(Outlier = ifelse(abs(CTNNB1) &gt; 3, &quot;Yes&quot;, &quot;No&quot;)) %&gt;%
  rownames_to_column(&quot;SIDM&quot;) %&gt;%
  left_join(data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;RNA&quot; = &quot;CTNNB1&quot;) %&gt;%
  rownames_to_column(&quot;SIDM&quot;), by = &quot;SIDM&quot;) %&gt;%
  left_join(data$prt_mtx %&gt;%
  mutate(Project_Identifier = gsub(&quot;;.*&quot;, &quot;&quot;, Project_Identifier)) %&gt;%
  column_to_rownames(&quot;Project_Identifier&quot;) %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Protein&quot;) %&gt;%
  mutate(&quot;Protein_name&quot; = gsub(&quot;;.*&quot;, &quot;&quot;, Protein)) %&gt;%
  mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = &quot;UNIPROT&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;%
  dplyr::select(-c(Protein_name, Protein)) %&gt;%
  dplyr::filter(Gene == &quot;CTNNB1&quot;) %&gt;%
  column_to_rownames(&quot;Gene&quot;) %&gt;%
  melt() %&gt;% set_colnames(c(&quot;SIDM&quot;, &quot;Protein&quot;))) %&gt;%
  ggplot(aes(x=RNA, y =Protein, color = Outlier)) + geom_point(size = 4) + theme_bw() + xlim(6,13) +
   scale_color_manual(values = c(&quot;No&quot; = &quot;#8d99ae&quot;,
                                &quot;Yes&quot; = &quot;#bc4749&quot;)) +
    theme(axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=20, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;none&quot;,
        strip.text.y = element_text(
        size = 10, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 10, family = &quot;serif&quot;
        )) + ggtitle(&quot;CTNNB1&quot;) + labs(x=&quot;RNA expression&quot;, y = &quot;Protein abundance&quot;)

ggsave(filename = &quot;output/Publication/Figures/Fig_3E_upper.png&quot;,height = 5.5, width = 5, plot = ctnnb1, dpi = 1200 )


 protein_swarm = data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;CTNNB1&quot;) %&gt;%
  scale() %&gt;%
  as.data.frame() %&gt;%
  mutate(Outlier = ifelse(abs(CTNNB1) &gt; 3, &quot;Yes&quot;, &quot;No&quot;)) %&gt;%
  rownames_to_column(&quot;SIDM&quot;) %&gt;%
  left_join(data$rna_mtx %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(&quot;RNA&quot; = &quot;CTNNB1&quot;) %&gt;%
  rownames_to_column(&quot;SIDM&quot;), by = &quot;SIDM&quot;) %&gt;%
  left_join(data$prt_mtx %&gt;%
  mutate(Project_Identifier = gsub(&quot;;.*&quot;, &quot;&quot;, Project_Identifier)) %&gt;%
  column_to_rownames(&quot;Project_Identifier&quot;) %&gt;%
  t() %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;Protein&quot;) %&gt;%
  mutate(&quot;Protein_name&quot; = gsub(&quot;;.*&quot;, &quot;&quot;, Protein)) %&gt;%
  mutate(&quot;Gene&quot; = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = &quot;UNIPROT&quot;, column = &quot;SYMBOL&quot;, multiVals=&quot;first&quot;)) %&gt;%
  dplyr::select(-c(Protein_name, Protein)) %&gt;%
  dplyr::filter(Gene == &quot;CTNNB1&quot;) %&gt;%
  column_to_rownames(&quot;Gene&quot;) %&gt;%
  melt() %&gt;% set_colnames(c(&quot;SIDM&quot;, &quot;Protein&quot;))) %&gt;%
  ggplot(aes(y = Protein, x=&#39;&#39;, color = Outlier)) +
  geom_beeswarm(size = 3, cex=3) +
scale_color_manual(values = c(&quot;No&quot; = &quot;#8d99ae&quot;,
                                &quot;Yes&quot; = &quot;#bc4749&quot;))  + theme_bw() +
    theme(axis.text.x = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
            axis.text.y = element_text(size = 20, family = &quot;serif&quot;, color = &quot;black&quot;),
        axis.title = element_text(size=20, family = &quot;serif&quot;),
         legend.box.background = element_rect(color = &quot;black&quot;),
          legend.text = element_text(family = &quot;serif&quot;),
          legend.title = element_text( family = &quot;serif&quot;),
           plot.title = element_text(family = &quot;serif&quot;, size =20),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = &quot;none&quot;,
        strip.text.y = element_text(
        size = 10, face = &quot;bold.italic&quot;, family = &quot;serif&quot;
        ),
         strip.text.x = element_text(
        size = 10, family = &quot;serif&quot;
        )) + ylab(&quot;Protein abundance&quot;) + xlab(&quot;Samples&quot;)


ggsave(filename = &quot;output/Publication/Figures/Fig_3E_lower.png&quot;,height = 5.5, width = 5, plot =protein_swarm, dpi = 1200 )</code></pre>
<pre class="r"><code>ggarrange(ctnnb1, protein_swarm, nrow = 2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/publication-figures.Rmd/unnamed-chunk-48-1.png" alt="Scatterplot showing mRNA expression and protein abundance for CTNNB1 (upper) and distribution of CTNNB1 protein abundance (lower) across samples in the ProCan-DepMapSanger dataset." width="576" />
<p class="caption">
Scatterplot showing mRNA expression and protein abundance for CTNNB1
(upper) and distribution of CTNNB1 protein abundance (lower) across
samples in the ProCan-DepMapSanger dataset.
</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Sydney
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggbeeswarm_0.7.2          openxlsx_4.2.7.1         
 [3] clusterProfiler_4.10.1    ggpubr_0.6.0             
 [5] matrixStats_1.5.0         msigdbr_7.5.1            
 [7] ragg_1.3.3                colorspace_2.1-1         
 [9] ggtext_0.1.2              VennDiagram_1.7.3        
[11] futile.logger_1.4.3       ggvenn_0.1.10            
[13] reshape2_1.4.4            ggiraph_0.8.12           
[15] svglite_2.1.3             EnsDb.Hsapiens.v86_2.99.0
[17] ensembldb_2.26.0          AnnotationFilter_1.26.0  
[19] GenomicFeatures_1.54.4    GenomicRanges_1.54.1     
[21] GenomeInfoDb_1.38.8       org.Hs.eg.db_3.18.0      
[23] AnnotationDbi_1.64.1      IRanges_2.36.0           
[25] S4Vectors_0.40.2          Biobase_2.62.0           
[27] BiocGenerics_0.48.1       RWDataPlyr_0.6.4         
[29] readxl_1.4.3              lubridate_1.9.4          
[31] forcats_1.0.0             stringr_1.5.1            
[33] dplyr_1.1.4               purrr_1.0.2              
[35] readr_2.1.5               tidyr_1.3.1              
[37] tibble_3.2.1              tidyverse_2.0.0          
[39] magrittr_2.0.3            ggplot2_3.5.1            
[41] workflowr_1.7.1          

loaded via a namespace (and not attached):
  [1] fs_1.6.5                    ProtGenerics_1.34.0        
  [3] bitops_1.0-9                enrichplot_1.22.0          
  [5] HDO.db_0.99.1               httr_1.4.7                 
  [7] RColorBrewer_1.1-3          tools_4.3.3                
  [9] backports_1.5.0             DT_0.33                    
 [11] R6_2.5.1                    lazyeval_0.2.2             
 [13] mgcv_1.9-1                  ggdist_3.3.2               
 [15] withr_3.0.2                 prettyunits_1.2.0          
 [17] gridExtra_2.3               cli_3.6.3                  
 [19] textshaping_0.4.1           formatR_1.14               
 [21] scatterpie_0.2.4            labeling_0.4.3             
 [23] sass_0.4.9                  Rsamtools_2.18.0           
 [25] systemfonts_1.1.0           yulab.utils_0.1.9          
 [27] gson_0.1.0                  DOSE_3.28.2                
 [29] rstudioapi_0.17.1           RSQLite_2.3.9              
 [31] generics_0.1.3              gridGraphics_0.5-1         
 [33] BiocIO_1.12.0               crosstalk_1.2.1            
 [35] zip_2.3.1                   car_3.1-3                  
 [37] distributional_0.5.0        GO.db_3.18.0               
 [39] Matrix_1.6-5                abind_1.4-8                
 [41] lifecycle_1.0.4             whisker_0.4.1              
 [43] yaml_2.3.10                 carData_3.0-5              
 [45] SummarizedExperiment_1.32.0 qvalue_2.34.0              
 [47] SparseArray_1.2.4           BiocFileCache_2.10.2       
 [49] blob_1.2.4                  promises_1.3.2             
 [51] crayon_1.5.3                lattice_0.22-5             
 [53] cowplot_1.1.3               KEGGREST_1.42.0            
 [55] pillar_1.10.1               knitr_1.49                 
 [57] fgsea_1.28.0                rjson_0.2.23               
 [59] codetools_0.2-19            fastmatch_1.1-6            
 [61] glue_1.8.0                  getPass_0.2-4              
 [63] ggfun_0.1.8                 data.table_1.16.4          
 [65] vctrs_0.6.5                 png_0.1-8                  
 [67] treeio_1.26.0               cellranger_1.1.0           
 [69] gtable_0.3.6                cachem_1.1.0               
 [71] xfun_0.50                   S4Arrays_1.2.1             
 [73] tidygraph_1.3.1             pheatmap_1.0.12            
 [75] nlme_3.1-164                ggtree_3.10.1              
 [77] bit64_4.5.2                 progress_1.2.3             
 [79] filelock_1.0.3              rprojroot_2.0.4            
 [81] bslib_0.8.0                 vipor_0.4.7                
 [83] DBI_1.2.3                   tidyselect_1.2.1           
 [85] processx_3.8.5              bit_4.5.0.1                
 [87] compiler_4.3.3              curl_6.1.0                 
 [89] git2r_0.35.0                xml2_1.3.6                 
 [91] DelayedArray_0.28.0         shadowtext_0.1.4           
 [93] rtracklayer_1.62.0          scales_1.3.0               
 [95] callr_3.7.6                 rappdirs_0.3.3             
 [97] digest_0.6.37               rmarkdown_2.29             
 [99] XVector_0.42.0              htmltools_0.5.8.1          
[101] pkgconfig_2.0.3             MatrixGenerics_1.14.0      
[103] dbplyr_2.5.0                fastmap_1.2.0              
[105] rlang_1.1.4                 htmlwidgets_1.6.4          
[107] farver_2.1.2                jquerylib_0.1.4            
[109] jsonlite_1.8.9              BiocParallel_1.36.0        
[111] GOSemSim_2.28.1             RCurl_1.98-1.16            
[113] Formula_1.2-5               GenomeInfoDbData_1.2.11    
[115] ggplotify_0.1.2             patchwork_1.3.0            
[117] munsell_0.5.1               Rcpp_1.0.14                
[119] ape_5.8-1                   babelgene_22.9             
[121] viridis_0.6.5               stringi_1.8.4              
[123] ggraph_2.2.1                zlibbioc_1.48.2            
[125] MASS_7.3-60.0.1             plyr_1.8.9                 
[127] parallel_4.3.3              ggrepel_0.9.6              
[129] Biostrings_2.70.3           graphlayouts_1.2.1         
[131] splines_4.3.3               gridtext_0.1.5             
[133] hms_1.1.3                   ps_1.8.1                   
[135] igraph_2.1.2                uuid_1.2-1                 
[137] ggsignif_0.6.4              biomaRt_2.58.2             
[139] futile.options_1.0.1        XML_3.99-0.18              
[141] evaluate_1.0.3              lambda.r_1.2.4             
[143] tzdb_0.4.0                  tweenr_2.0.3               
[145] httpuv_1.6.15               polyclip_1.10-7            
[147] ggforce_0.4.2               broom_1.0.7                
[149] restfulr_0.0.15             tidytree_0.4.6             
[151] rstatix_0.7.2               later_1.4.1                
[153] viridisLite_0.4.2           aplot_0.2.4                
[155] beeswarm_0.4.0              memoise_2.0.1              
[157] GenomicAlignments_1.38.2    timechange_0.3.0           
[159] here_1.0.1                 </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
