---
title: "Correlation analysis "
author: "urwahnawaz"
date: "2025-02-27"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---



```{r setup, include=FALSE}
knitr::opts_chunk$set(
  eval = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE,
  cache=TRUE
)
```

## Introduction 

This notebook contains the code that calculates mRNA-protein correlations for the 949 ProCan-DepMapSanger cancer cell line data. In addition, it also contains code which calculates correlations for the CPTAC data retrieved from LinkedOmics KB, and CCLE datasets. 

We have calculated both correlations of the entire dataset, and tissue specific correlations. 

### Libraries used 

```{r}
library(ggplot2)
library(magrittr)
library(tidyverse)
library(readxl)
library(dplyr)
library(RWDataPlyr)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(EnsDb.Hsapiens.v86)
library(svglite)
library(ggiraph)
library(reshape2)
library(ggvenn)
library(VennDiagram)
library(ggtext)
library(colorspace)
library(ragg)
library(msigdbr)
library(matrixStats)
library(ggiraph)
```

## ProCan-DepMapSanger Dataset 

### Load data 

```{r}
file_dir = file.path("/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/")
data = list()
data$sample_md = read.delim("~/Documents/2023/RNA-protein-correlationx/Data/e0022_diann_051021_sample_mapping_averaged.txt")
data$sample_md_rep = read.table(file.path(file_dir, "e0022_diann_051021_sample_mapping_replicates.txt"), header=TRUE, sep = "\t")

data$prt_mtx =  read.delim(file.path(file_dir, "e0022_diann_051021_working_matrix_averaged.txt"), 
                           header = TRUE, fill = TRUE, check.names = FALSE) 
data$rna_voom = read.csv(gzfile(file.path(file_dir, "rnaseq_voom.csv.gz"))) %>%
  dplyr::rename("Gene" = "X")
uniprot_annot = read.table(file.path(file_dir, "idmapping_2024_02_13.tsv"), 
                           sep = "\t", fill = TRUE, header=TRUE, quote = "") %>% 
  mutate("Protein_name" = paste0(Entry, ";", `Entry.Name`)) %>% 
  distinct(Protein_name, .keep_all = TRUE) %>% 
  dplyr::select(Protein_name, Gene = `Gene.Names`) %>% 
   mutate(Gene = str_split(Gene, " ")) %>%
  unnest(Gene)
```

```{r}
data$rna_voom %<>% 
  left_join(uniprot_annot , by = "Gene") 
```

### Calculate overall correlations 


```{r}
data_full = data$prt_mtx %>% 
  as.data.frame() %>% 
  column_to_rownames("Project_Identifier") %>%
  t() %>% 
  as.data.frame() %>%
  rownames_to_column("Protein") %>%
 mutate("Protein_name" = gsub(";.*", "", Protein)) %>%
  mutate("Gene" = mapIds(org.Hs.eg.db, keys = Protein_name,
                         keytype = "UNIPROT", column = "SYMBOL", multiVals="first"))



```

```{r}
cor_all= list()
cell_lines = data$sample_md %>% 
    dplyr::select(Project_Identifier, SIDM)

```

Proteins were considered if they had > 10 observations for them in each RNA and proteomic datasets. 

```{r message=FALSE, warning=FALSE}
for (prt in data_full$Gene) {

 
    cor_df = data$rna_voom %>% 
      dplyr::filter(Gene %in% prt) %>% 
      melt() %>% dplyr::rename("SIDM" = "variable", "RNA" = "value") %>% 
      left_join(cell_lines, by ="SIDM") %>% 
      dplyr::select("Cell-lines" = "Project_Identifier", 
                    RNA) %>%
      full_join(data_full %>% 
        dplyr::filter(Gene %in% prt) %>% 
          
      dplyr::select(contains(cell_lines$Project_Identifier)) %>%
        melt() %>% dplyr::rename("Cell-lines" = "variable", "Protein" = "value"), 
      by = "Cell-lines") %>% drop_na()
    
      if(nrow(cor_df) > 10) { 

      cor_all[[paste0(prt)]]  = cor_df %>% 
       with(cor(.$RNA,.$Protein, method = "spearman")) %>% 
        as.data.frame() %>% 
        set_colnames("Correlation") %>% 
        mutate(Protein = prt)

    }
    
}

corAll = cor_all %>%
  do.call(rbind,.) %>% 
  drop_na()



#write.csv(corAll, "output/correlations-datasets/correlation-all-nontransformed.csv")
```

```{r}
corAll  %>%
  ggplot(aes(x = Correlation)) + 
  geom_histogram(color="white", alpha=1, fill = "#ccd5ae") + theme_bw() +
  labs(x="mRNA-protein correlation", 
       y = "Number of associations") +  annotate("text", x = -0.1, y = 600,
  label = "paste(italic(median), \" = 0.42\")", parse = TRUE, color = "#9a031e") + 
  geom_vline(xintercept = 0.42, linetype = "dashed", color = "#9a031e") +
  ggtitle("ProCan-DepMapSanger") +
     theme(legend.position = "none",
      axis.text=element_text(size=15,  family="serif", color = "black"),
      axis.title = element_text(size=16, family = "serif", color = "black")) 

```

### Correlations for each tissue 

Selecting tissues which contain more than 10 cell lines. Subsequently, only calculating correlations for proteins which contain more than 10 observaions for both RNA and protein abundances. 

```{r fig.height =6, fig.width=6}

tissue_counts <-table(data$sample_md$Tissue_type)
tissues_incl <- names(tissue_counts[tissue_counts > 10])

tissue_counts %>% 
  as.data.frame() %>%
  mutate(col = ifelse(Freq >10, "Keep", "Not Keep")) %>%
  ggplot(aes(x=reorder(Var1, Freq), y=Freq, fill = col)) + geom_bar(stat ="identity")  + theme_bw() +
  coord_flip() + 
  labs(x="Tissues", y ="Number of samples", 
       title = "Tissues") +
   theme(legend.position = "none",
      axis.text=element_text(size=10,  family="serif", color = "black"),
      axis.title = element_text(size=12, family = "serif", color = "black")) + 
  scale_fill_manual(values = c("Keep" = "#8c2f39",
                               "Not Keep "))
```

```{r}
cor_per_tissue = list()


for (tis in tissues_incl){
  
  message(paste("Now performing correlation analysis for", tis))
  cell_lines = data$sample_md %>% 
    dplyr::filter(Tissue_type ==  tis ) %>% 
    dplyr::select(Project_Identifier, SIDM)
  
  cor_analysis = list()
  
  for (prt in data_full$Gene) {
 
    cor_df = data$rna_voom %>% 
      dplyr::filter(Gene %in% prt) %>% 
      dplyr::select(contains(cell_lines$SIDM)) %>%
      melt() %>% dplyr::rename("SIDM" = "variable", "RNA" = "value") %>% 
      left_join(cell_lines, by ="SIDM") %>% 
      dplyr::select("Cell-lines" = "Project_Identifier", 
                    RNA) %>%
      full_join(data_full%>% 
                  dplyr::filter(Gene %in% prt)  %>% 
                  dplyr::select(contains(cell_lines$Project_Identifier)) %>%
                  melt() %>% 
                  dplyr::rename("Cell-lines" = "variable", "Protein" = "value"),
      by = "Cell-lines") %>% drop_na()
    
      if(nrow(cor_df) > 10) { 

      cor_per_tissue[[paste0(tis,"_", prt)]]  = cor_df %>% 
       with(cor(.$RNA,.$Protein, method = "spearman")) %>% 
        as.data.frame() %>% 
        set_colnames("Correlation") %>% 
        mutate(Tissue = tis, 
               Protein = prt)

    }
    
  }

}


corTissue = cor_per_tissue  %>%
  do.call(rbind,.) %>% 
  drop_na() %>% 
  dplyr::rename(Gene = Protein) %>% 
  left_join(data_full %>% 
              dplyr::select(Gene, Protein), by = "Gene")




#prev_tis_cor = read.csv("output/correlation-tissue-nontransformed.csv")

#prev_tis_cor %>% DT::datatable()
#write.csv(corTissue, "output/correlations-datasets/correlation-tissue-nontransformed.csv")
```


## CPTAC data correlation

```{r}
subfolders = list.dirs(file.path("/Users/unawaz/Documents/2024/Correlations/CPTAC"), full.names = TRUE)
rna_file = list.files(subfolders, pattern = "RSEM", full.names = TRUE)
ptn_file = list.files(subfolders, pattern = "normalized_Tumor.txt", full.names = TRUE)
metadata  =  list.files(subfolders, pattern = "meta.txt", full.names = TRUE)
RNA_seq = list()

for (rna in rna_file){
  dataset_name = basename(dirname(rna))
  RNA_seq[[paste0(dataset_name)]] = read.table(rna, header=TRUE, check.names  = FALSE)

}


ptx = list()
for (ptn in ptn_file){
  dataset_name = basename(dirname(ptn))
  ptx[[paste0(dataset_name)]] = read.table(ptn, header=TRUE, check.names  = FALSE, row.names = 1)

}

md = list()

for (met in metadata){
  dataset_name = basename(dirname(met))
  md[[paste0(dataset_name)]] = read.table(met, check.names  = FALSE, fill = TRUE, header=TRUE, sep = "\t")

}
```

We merged all the RNA and protein matrices into single data list for each type of abundance across all cancers.


```{r}
cancers = names(ptx)

all_data = ptx$BRCA %>%
  rownames_to_column("Gene")
for (cancer in cancers[-1]){

  all_data %<>%
    full_join(ptx[[cancer]] %>%
                rownames_to_column("Gene"), by = "Gene")
}


#write.csv(all_data, "output/CPTAC-proteomics-merged.csv")

all_data_rna = RNA_seq$BRCA

for (cancer in cancers[-1]){

  all_data_rna %<>%
    full_join(RNA_seq[[cancer]], by = "idx")
}


#write.csv(all_data_rna, "output/CPTAC-RNAseq-merged.csv")


library(data.table)
rbindlist(md, fill = TRUE)

all_md = rbindlist(md, fill = TRUE) %>% as.data.frame()
#write.csv(all_md, "output/CPTAC-sample-info-merged.csv")



for (cancer in cancers){

  md[[cancer]] %<>%
    mutate(Cancer = cancer)
}
```

### Overall correlations

Similar to the ProCan-DepMapSanger dataset, we performed correlations across the entire dataset, where only proteins with > 10 observations in both the protein matrix and the RNA matrix were considered for analysis.


```{r}
cor_all = list()

for (protein in all_data$Gene) {


cor_df =  all_data_rna %>%
      dplyr::filter(idx %in% protein) %>%
      melt() %>% dplyr::rename("SIDM" = "variable", "RNA" = "value") %>%
      full_join(all_data %>%
                   dplyr::filter(Gene %in% protein) %>%
                  melt() %>%
      dplyr::rename("SIDM" = "variable", "Protein" = "value") %>%
        dplyr::select(SIDM, Protein),
      by = "SIDM") %>% drop_na()

      if(nrow(cor_df) > 10) {

      cor_all[[paste0(protein)]]  = cor_df %>%
       with(cor(.$RNA,.$Protein, method = "spearman")) %>%
        as.data.frame() %>%
        set_colnames("Correlation") %>%
        mutate(
               Protein = protein)

    }

}


cptac=  cor_all %>%
  do.call(rbind,.)


median(cptac$Correlation, na.rm =T )

```


### Correlation per tissue type

```{r}
cor_per_cancer = list()
cancers = names(ptx)

for (cancer in cancers){
  ptx_matrix = ptx[[cancer]]
  rna_matrix = RNA_seq[[cancer]]


   for (protein in rownames(ptx_matrix)) {


    cor_df = rna_matrix %>%
      dplyr::filter(idx %in% protein) %>%
      melt() %>% dplyr::rename("SIDM" = "variable", "RNA" = "value") %>%
      full_join(ptx_matrix %>%
                  rownames_to_column("Genes") %>%
                   dplyr::filter(Genes %in% protein) %>%
                  melt() %>%
      dplyr::rename("SIDM" = "variable", "Protein" = "value") %>%
        dplyr::select(SIDM, Protein),
      by = "SIDM") %>% drop_na()

      if(nrow(cor_df) > 10) {

      cor_per_cancer[[paste0(cancer,"_", protein)]]  = cor_df %>%
       with(cor(.$RNA,.$Protein, method = "spearman")) %>%
        as.data.frame() %>%
        set_colnames("Correlation") %>%
        mutate(Cancer = cancer,
               Protein = protein)

    }

  }
}

cptac= cor_per_cancer %>%
  do.call(rbind,.)
```


## CCLE data

```{r}
CCLE_data = read_excel("/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/mmc4.xlsx",
                       sheet =2)
```


```{r}
data_ccle = list()

data_ccle$prt = read.csv("/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/protein_quant_current_normalized.csv", header=TRUE)
data_ccle$rna = read.csv("/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/CCLE_expression.csv",
                         header=TRUE, check.names = FALSE)

```


```{r}
data_ccle$sample_md = read.csv("/Users/unawaz/Documents/2023/RNA-protein-correlationx/Data/CCLE/sample_info.csv", header=TRUE,
                               check.names = FALSE)
```


## Fix metadata for tissues

```{r}
ccle_tissue_annot = read_excel("data/Datasets/CCLE/CCLE_tissue_names.xlsx") %>%
  as.data.frame()
```

```{r}
data_ccle$sample_md %<>%
  mutate(Lineage = gsub("_", " ", lineage)) %>%
  mutate(Lineage = str_to_title(Lineage)) %>%
  left_join(ccle_tissue_annot, by = "Lineage")
```

```{r}
colnames(data_ccle$rna)[1] = c("DepMap_ID")

data_ccle$rna %<>%
  column_to_rownames("DepMap_ID") %>%
  t() %>%
  as.data.frame() %>%
  rownames_to_column("Gene") %>%
  mutate(Gene = gsub(" .*", "", Gene))  %>%
  mutate("Protein" = mapIds(org.Hs.eg.db, keys = Gene,
                         column = "UNIPROT", keytype= "SYMBOL", multiVals="first")) %>%
  drop_na(Protein)


data_ccle$prt = data_ccle$prt[,-c(7:48)]


data_ccle$prt %<>%
  dplyr::select(-c(Protein_Id, Description,
                "Group_ID","Uniprot")) %>%
  column_to_rownames("Uniprot_Acc")

colnames(data_ccle$prt) = gsub("_Ten.*", "",colnames(data_ccle$prt))
```

```{r}
library(ggvenn)
library(VennDiagram)

x = list("Protein" = data_ccle$prt$Gene_Symbol,
         "Gene" = data_ccle$rna$Gene)

ggvenn(x)

lists = calculate.overlap(x)

lists$a3
```

### Correlation across the dataset

```{r}
ccle_cor_all  = list()



  prt_data = data_ccle$prt %>%
  melt() %>%
  dplyr::rename(CCLE_Name = variable, Protein = value) %>%
  inner_join(data_ccle$sample_md %>%
               dplyr::select(CCLE_Name, DepMap_ID,  `Tissue_type`= `New Lineage`), by = "CCLE_Name") %>%
  dplyr::select(-c(CCLE_Name))

 rna_data = data_ccle$rna %>%
   melt() %>%
   dplyr::select(-Protein) %>%
   dplyr::rename(DepMap_ID = variable, "RNA" = value)

 ccle_exp_data = rna_data %>%
   mutate(identifier = paste0(Gene, "_", DepMap_ID)) %>%
   distinct(identifier, .keep_all = TRUE) %>%
   dplyr::select(- DepMap_ID) %>%
   left_join(prt_data %>%
               mutate(identifier = paste0(Gene_Symbol, "_", DepMap_ID)),  by = "identifier") %>%
   drop_na() %>%
   dplyr::select(-Gene_Symbol) %>%
   dplyr::select(Gene, SIDM= DepMap_ID, RNA, Protein, Tissue_type) %>%
   mutate(Dataset = c("CCLE"),
          Tissue_type = ifelse(Tissue_type == "Lubg", "Lung", Tissue_type))

  for (prt in lists$a3) {


    cor_data= rna_data %>%
      dplyr::filter(Gene == prt) %>%
      full_join(prt_data %>%
                  dplyr::filter(Gene_Symbol == prt) %>%
                  dplyr::select(-Gene_Symbol),
                by = "DepMap_ID") %>%
      drop_na()

      if(nrow(cor_data) > 10) {

       ccle_cor_all[[paste(prt)]]  = cor_data %>%
         with(cor(.$Protein, .$RNA, method = "spearman")) %>%
         as.data.frame() %>%
        set_colnames("Correlation") %>%
       mutate(Protein = prt)

    }

  }

 ccle_cor_all %<>% do.call(rbind,.) %>%  drop_na()

 median(ccle_cor_all$Correlation)
```


### Tissue type specifc

```{r}
cancers = table(data_ccle$sample_md$primary_disease) %>%
   melt() %>%
   dplyr::filter(value >10) %>%
   set_colnames("Cancer") %>%
   as.data.frame()
#
# sample_cancers = data_ccle$sample_md %>%
#   dplyr::filter(disease %in% cancers$Cancer)


tissues= colnames(data_ccle$prt) %>%
  as.data.frame() %>%
  set_names(c("CCLE_Name")) %>%
  inner_join(data_ccle$sample_md, by = "CCLE_Name") %>%
  with(table(.$`New Lineage`)) %>%
  melt() %>%
  dplyr::filter(value >10) %>%
  set_colnames(c("Tissue", "Number"))

ccle_cor = list()
for (cancer in tissues$Tissue) {
  samples =  data_ccle$sample_md %>%
  dplyr::filter(`New Lineage` %in% cancer)

  rna_data = data_ccle$rna %>%
    dplyr::select(Gene, contains(samples$DepMap_ID))

  prt_data = data_ccle$prt %>%
    dplyr::select(Gene_Symbol, any_of(samples$CCLE_Name)) %>%
  melt() %>%
  dplyr::rename(CCLE_Name = variable, Protein = value) %>%
  inner_join(samples %>%
               dplyr::select(CCLE_Name, DepMap_ID), by = "CCLE_Name") %>%
  dplyr::select(-c(CCLE_Name))

 rna_data = data_ccle$rna %>%
    dplyr::select(Gene, any_of(samples$DepMap_ID)) %>%
   melt() %>%
   dplyr::rename(DepMap_ID = variable, "RNA" = value)

  for (prt in lists$a3) {


    cor_data= rna_data %>%
      dplyr::filter(Gene == prt) %>%
      full_join(prt_data %>%
                  dplyr::filter(Gene_Symbol == prt) %>%
                  dplyr::select(-Gene_Symbol),
                by = "DepMap_ID") %>%
      drop_na()

      if(nrow(cor_data) > 10) {

       ccle_cor[[paste(prt, cancer)]]  = cor_data %>%
         with(cor(.$Protein, .$RNA, method = "spearman")) %>%
         as.data.frame() %>%
        set_colnames("Correlation") %>%
       mutate(Tissue = cancer,
             Protein = prt)

    }

  }

}

ccle_cor %<>% do.call(rbind,.) %>%  drop_na()

table(ccle_cor$Tissue)


```

